{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Introduction\n",
    "This notebook describes how the data clean is performed to organise curated data ready for database\n",
    "\n",
    "This notebook requires kernels python3 and R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step1. get the variation_proportion and depth column from each curated files  (Kernel: python3)\n",
    "input file: previous curated data\n",
    "\n",
    "output file: curated data only with column ID, chr, position, variation proportion, depth, deletionCount and deletionProportion in dir 'clean_data_selected_column'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "data_dir = os.path.join(\"../modification_tcga/clean_completed_v2/clean_completed_final\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9834\n",
      "11_10821794.completed\n",
      "9833\n"
     ]
    }
   ],
   "source": [
    "# get the file names \n",
    "data_dir_doc = os.listdir(data_dir)\n",
    "print(len(data_dir_doc))\n",
    "print(data_dir_doc[0])\n",
    "\n",
    "# remove files that do not end with .completed\n",
    "data_dir_doc2 = [filename for filename in data_dir_doc if filename.find(\".completed\")>=0]\n",
    "print(len(data_dir_doc2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../modification_tcga/clean_completed_v2/clean_completed_final/11_10821794.completed\n"
     ]
    }
   ],
   "source": [
    "data_dir_doc_full = [os.path.join(data_dir, filename)\n",
    "                    for filename in data_dir_doc2]\n",
    "print(data_dir_doc_full[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "5_151189490.completed don't have selected columns.\n",
      "500\n",
      "19_19103949.completed don't have selected columns.\n",
      "1000\n",
      "1500\n",
      "sites.unfinished.final_.completed don't have selected columns.\n",
      "2000\n",
      "1_168165829.completed don't have selected columns.\n",
      "2500\n",
      "3000\n",
      "3500\n",
      "4000\n",
      "4500\n",
      "5000\n",
      "nohup.out_.completed don't have selected columns.\n",
      "14_31570413.completed don't have selected columns.\n",
      "5500\n",
      "17_61908910.completed don't have selected columns.\n",
      "6000\n",
      "6500\n",
      "7000\n",
      "7500\n",
      "8000\n",
      "8500\n",
      "9000\n",
      "9500\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(data_dir_doc2)):\n",
    "    filename_i = data_dir_doc2[i]\n",
    "    file_dir_i = data_dir_doc_full[i]\n",
    "    \n",
    "    read_file_1 = pd.read_csv(file_dir_i, sep=\"\\t\")\n",
    "    target_column = [\"analysis_ID\", \"chrom\", \"position\",\"reference_nt\", \n",
    "                     \"variant_proportion\", \"depth\", \"deletionCount\", \n",
    "                     \"deletion_proportion\"]\n",
    "    if all(a in list(read_file_1.columns.values) for a in target_column):\n",
    "        read_file_1_sub = read_file_1.loc[:, target_column]\n",
    "        read_file_1_sub.to_csv(os.path.join(\"./clean_data_selected_column\", filename_i), sep=\"\\t\", \n",
    "                header=True, index=False)\n",
    "    else:\n",
    "        print(filename_i+\" don't have selected columns.\")\n",
    "    if i%500 == 0:\n",
    "        print(i)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step2. refer to the 2_seggregate_RNA_mod_loci_for_each_sample.bash"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step3. Get the information for each analysis ID (Kernel: python3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11336\n",
      "[36]\n"
     ]
    }
   ],
   "source": [
    "# list all unqiue IDs\n",
    "all_ids = os.listdir(\"./RNA_mod_for_each_sample/\")\n",
    "print(len(all_ids))\n",
    "\n",
    "# count the length for each id\n",
    "all_ids_len = [len(each_id) for each_id in all_ids]\n",
    "all_ids_len_np = np.array(all_ids_len)\n",
    "print(np.unique(all_ids_len_np))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11336\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "# number of ids equal to 11 or 36 in length\n",
    "print(len(all_ids_len_np[all_ids_len_np==36]))\n",
    "print(len(all_ids_len_np[all_ids_len_np==11]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11336\n",
      "6adb30fd-fe7f-45fa-9fd4-81d4bd11dd65\n"
     ]
    }
   ],
   "source": [
    "# id with length 11 is not true id and should be removed\n",
    "all_ids_final = [id_i for id_i in all_ids if len(id_i)==36]\n",
    "print(len(all_ids_final))\n",
    "print(all_ids_final[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study</th>\n",
       "      <th>barcode</th>\n",
       "      <th>disease</th>\n",
       "      <th>disease_name</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>sample_type_name</th>\n",
       "      <th>analyte_type</th>\n",
       "      <th>library_type</th>\n",
       "      <th>center</th>\n",
       "      <th>center_name</th>\n",
       "      <th>...</th>\n",
       "      <th>aliquot_id</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>sample_accession</th>\n",
       "      <th>published</th>\n",
       "      <th>uploaded</th>\n",
       "      <th>modified</th>\n",
       "      <th>state</th>\n",
       "      <th>reason</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA</td>\n",
       "      <td>TCGA-E9-A1NH-01A-11D-A14G-09</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>Breast invasive carcinoma</td>\n",
       "      <td>TP</td>\n",
       "      <td>01</td>\n",
       "      <td>DNA</td>\n",
       "      <td>WGS</td>\n",
       "      <td>WUGSC</td>\n",
       "      <td>WUGSC</td>\n",
       "      <td>...</td>\n",
       "      <td>13c312ec-0add-4758-ab8d-c193e2e08c6d</td>\n",
       "      <td>b2aac45b-2073-4c7a-adb9-769a4fdcc111</td>\n",
       "      <td>fb0b2d84-d982-4d3a-9803-36420f2e7e46</td>\n",
       "      <td>E9</td>\n",
       "      <td></td>\n",
       "      <td>2013-10-22</td>\n",
       "      <td>2013-10-21</td>\n",
       "      <td>2014-06-19</td>\n",
       "      <td>Live</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA</td>\n",
       "      <td>TCGA-66-2763-01A-01T-1557-13</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>Lung squamous cell carcinoma</td>\n",
       "      <td>TP</td>\n",
       "      <td>01</td>\n",
       "      <td>Total RNA</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>BCCAGSC</td>\n",
       "      <td>BCCAGSC</td>\n",
       "      <td>...</td>\n",
       "      <td>13c3399e-87e9-4213-a8d8-529c3817869f</td>\n",
       "      <td>1e3eb77c-d85b-437e-8b8c-12e5bd0e93dc</td>\n",
       "      <td>a7d715ab-8ecd-4a25-9035-97aeba7a24c0</td>\n",
       "      <td>66</td>\n",
       "      <td>SRS229695</td>\n",
       "      <td>2011-06-13</td>\n",
       "      <td>2011-06-07</td>\n",
       "      <td>2013-05-16</td>\n",
       "      <td>Live</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA</td>\n",
       "      <td>TCGA-66-2763-01A-01T-1557-13</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>Lung squamous cell carcinoma</td>\n",
       "      <td>TP</td>\n",
       "      <td>01</td>\n",
       "      <td>Total RNA</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>BCCAGSC</td>\n",
       "      <td>BCCAGSC</td>\n",
       "      <td>...</td>\n",
       "      <td>13c3399e-87e9-4213-a8d8-529c3817869f</td>\n",
       "      <td>1e3eb77c-d85b-437e-8b8c-12e5bd0e93dc</td>\n",
       "      <td>a7d715ab-8ecd-4a25-9035-97aeba7a24c0</td>\n",
       "      <td>66</td>\n",
       "      <td>SRS229695</td>\n",
       "      <td>2013-05-02</td>\n",
       "      <td>2013-05-02</td>\n",
       "      <td>2013-05-16</td>\n",
       "      <td>Live</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA</td>\n",
       "      <td>TCGA-DQ-5630-01A-01R-1873-07</td>\n",
       "      <td>HNSC</td>\n",
       "      <td>Head and Neck squamous cell carcinoma</td>\n",
       "      <td>TP</td>\n",
       "      <td>01</td>\n",
       "      <td>RNA</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>...</td>\n",
       "      <td>13c839d1-f77b-4bde-9bd5-8d5165d94346</td>\n",
       "      <td>91a712b5-a724-48d0-ba06-4f6857683463</td>\n",
       "      <td>a0cdb208-e66f-4ad7-86fa-5077145a4a68</td>\n",
       "      <td>DQ</td>\n",
       "      <td></td>\n",
       "      <td>2013-09-27</td>\n",
       "      <td>2013-09-25</td>\n",
       "      <td>2013-09-27</td>\n",
       "      <td>Live</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA</td>\n",
       "      <td>TCGA-DQ-5630-01A-01R-1873-07</td>\n",
       "      <td>HNSC</td>\n",
       "      <td>Head and Neck squamous cell carcinoma</td>\n",
       "      <td>TP</td>\n",
       "      <td>01</td>\n",
       "      <td>RNA</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>...</td>\n",
       "      <td>13c839d1-f77b-4bde-9bd5-8d5165d94346</td>\n",
       "      <td>91a712b5-a724-48d0-ba06-4f6857683463</td>\n",
       "      <td>a0cdb208-e66f-4ad7-86fa-5077145a4a68</td>\n",
       "      <td>DQ</td>\n",
       "      <td></td>\n",
       "      <td>2012-09-21</td>\n",
       "      <td>2012-09-21</td>\n",
       "      <td>2013-05-16</td>\n",
       "      <td>Live</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  study                       barcode disease  \\\n",
       "0  TCGA  TCGA-E9-A1NH-01A-11D-A14G-09    BRCA   \n",
       "1  TCGA  TCGA-66-2763-01A-01T-1557-13    LUSC   \n",
       "2  TCGA  TCGA-66-2763-01A-01T-1557-13    LUSC   \n",
       "3  TCGA  TCGA-DQ-5630-01A-01R-1873-07    HNSC   \n",
       "4  TCGA  TCGA-DQ-5630-01A-01R-1873-07    HNSC   \n",
       "\n",
       "                            disease_name sample_type sample_type_name  \\\n",
       "0              Breast invasive carcinoma          TP               01   \n",
       "1           Lung squamous cell carcinoma          TP               01   \n",
       "2           Lung squamous cell carcinoma          TP               01   \n",
       "3  Head and Neck squamous cell carcinoma          TP               01   \n",
       "4  Head and Neck squamous cell carcinoma          TP               01   \n",
       "\n",
       "  analyte_type library_type    center center_name  ...  \\\n",
       "0          DNA          WGS     WUGSC       WUGSC  ...   \n",
       "1    Total RNA    miRNA-Seq   BCCAGSC     BCCAGSC  ...   \n",
       "2    Total RNA    miRNA-Seq   BCCAGSC     BCCAGSC  ...   \n",
       "3          RNA      RNA-Seq  UNC-LCCC    UNC-LCCC  ...   \n",
       "4          RNA      RNA-Seq  UNC-LCCC    UNC-LCCC  ...   \n",
       "\n",
       "                             aliquot_id                        participant_id  \\\n",
       "0  13c312ec-0add-4758-ab8d-c193e2e08c6d  b2aac45b-2073-4c7a-adb9-769a4fdcc111   \n",
       "1  13c3399e-87e9-4213-a8d8-529c3817869f  1e3eb77c-d85b-437e-8b8c-12e5bd0e93dc   \n",
       "2  13c3399e-87e9-4213-a8d8-529c3817869f  1e3eb77c-d85b-437e-8b8c-12e5bd0e93dc   \n",
       "3  13c839d1-f77b-4bde-9bd5-8d5165d94346  91a712b5-a724-48d0-ba06-4f6857683463   \n",
       "4  13c839d1-f77b-4bde-9bd5-8d5165d94346  91a712b5-a724-48d0-ba06-4f6857683463   \n",
       "\n",
       "                              sample_id tss_id  sample_accession   published  \\\n",
       "0  fb0b2d84-d982-4d3a-9803-36420f2e7e46     E9                    2013-10-22   \n",
       "1  a7d715ab-8ecd-4a25-9035-97aeba7a24c0     66         SRS229695  2011-06-13   \n",
       "2  a7d715ab-8ecd-4a25-9035-97aeba7a24c0     66         SRS229695  2013-05-02   \n",
       "3  a0cdb208-e66f-4ad7-86fa-5077145a4a68     DQ                    2013-09-27   \n",
       "4  a0cdb208-e66f-4ad7-86fa-5077145a4a68     DQ                    2012-09-21   \n",
       "\n",
       "     uploaded    modified state reason  \n",
       "0  2013-10-21  2014-06-19  Live         \n",
       "1  2011-06-07  2013-05-16  Live         \n",
       "2  2013-05-02  2013-05-16  Live         \n",
       "3  2013-09-25  2013-09-27  Live         \n",
       "4  2012-09-21  2013-05-16  Live         \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the information files containing analysis id\n",
    "all_info = pd.read_csv(\"tcga_ccle_samples.barcode.txt\", sep=\"\\t\",low_memory=False)\n",
    "all_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['study', 'barcode', 'disease', 'disease_name', 'sample_type', 'sample_type_name', 'analyte_type', 'library_type', 'center', 'center_name', 'platform', 'platform_name', 'assembly', 'filename', 'files_size', 'checksum', 'analysis_id', 'aliquot_id', 'participant_id', 'sample_id', 'tss_id', 'sample_accession', 'published', 'uploaded', 'modified', 'state', 'reason']\n"
     ]
    }
   ],
   "source": [
    "print(list(all_info.columns.values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11336\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study</th>\n",
       "      <th>barcode</th>\n",
       "      <th>disease</th>\n",
       "      <th>disease_name</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>sample_type_name</th>\n",
       "      <th>analyte_type</th>\n",
       "      <th>library_type</th>\n",
       "      <th>center</th>\n",
       "      <th>center_name</th>\n",
       "      <th>...</th>\n",
       "      <th>aliquot_id</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>sample_accession</th>\n",
       "      <th>published</th>\n",
       "      <th>uploaded</th>\n",
       "      <th>modified</th>\n",
       "      <th>state</th>\n",
       "      <th>reason</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA</td>\n",
       "      <td>TCGA-DQ-5630-01A-01R-1873-07</td>\n",
       "      <td>HNSC</td>\n",
       "      <td>Head and Neck squamous cell carcinoma</td>\n",
       "      <td>TP</td>\n",
       "      <td>01</td>\n",
       "      <td>RNA</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>...</td>\n",
       "      <td>13c839d1-f77b-4bde-9bd5-8d5165d94346</td>\n",
       "      <td>91a712b5-a724-48d0-ba06-4f6857683463</td>\n",
       "      <td>a0cdb208-e66f-4ad7-86fa-5077145a4a68</td>\n",
       "      <td>DQ</td>\n",
       "      <td></td>\n",
       "      <td>2013-09-27</td>\n",
       "      <td>2013-09-25</td>\n",
       "      <td>2013-09-27</td>\n",
       "      <td>Live</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TCGA</td>\n",
       "      <td>TCGA-KK-A7B0-01A-11R-A32O-07</td>\n",
       "      <td>PRAD</td>\n",
       "      <td>Prostate adenocarcinoma</td>\n",
       "      <td>TP</td>\n",
       "      <td>01</td>\n",
       "      <td>RNA</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>...</td>\n",
       "      <td>13d2199f-d552-4ea3-9d7c-c2d1fb01d7a7</td>\n",
       "      <td>73d4e806-2c15-4c9e-a978-106fb9494e82</td>\n",
       "      <td>e93cc6cc-ce3b-45eb-8928-d5b13d7194af</td>\n",
       "      <td>KK</td>\n",
       "      <td></td>\n",
       "      <td>2014-01-17</td>\n",
       "      <td>2014-01-17</td>\n",
       "      <td>2014-01-17</td>\n",
       "      <td>Live</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TCGA</td>\n",
       "      <td>TCGA-DD-A3A3-11A-11R-A22L-07</td>\n",
       "      <td>LIHC</td>\n",
       "      <td>Liver hepatocellular carcinoma</td>\n",
       "      <td>NT</td>\n",
       "      <td>11</td>\n",
       "      <td>RNA</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>...</td>\n",
       "      <td>13d4c229-cf05-4fca-a52d-798dc7894ba1</td>\n",
       "      <td>516c8594-2973-41e5-a5ce-76d177edfa48</td>\n",
       "      <td>1079b0a9-f74e-4d0a-bed0-22aad6ea64e2</td>\n",
       "      <td>DD</td>\n",
       "      <td></td>\n",
       "      <td>2013-09-21</td>\n",
       "      <td>2013-09-21</td>\n",
       "      <td>2013-09-21</td>\n",
       "      <td>Live</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TCGA</td>\n",
       "      <td>TCGA-VV-A829-01A-21R-A36H-07</td>\n",
       "      <td>LGG</td>\n",
       "      <td>Brain Lower Grade Glioma</td>\n",
       "      <td>TP</td>\n",
       "      <td>01</td>\n",
       "      <td>RNA</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>...</td>\n",
       "      <td>13d82ac1-eecc-41bb-af05-4fd4f614f260</td>\n",
       "      <td>10887e42-444d-49fc-aee5-dd0e9fc4ea54</td>\n",
       "      <td>9f00d856-b402-46d2-9bb7-2e9e4c2ca19c</td>\n",
       "      <td>VV</td>\n",
       "      <td></td>\n",
       "      <td>2014-06-24</td>\n",
       "      <td>2014-06-20</td>\n",
       "      <td>2014-06-24</td>\n",
       "      <td>Live</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA</td>\n",
       "      <td>TCGA-56-7222-01A-11R-2045-07</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>Lung squamous cell carcinoma</td>\n",
       "      <td>TP</td>\n",
       "      <td>01</td>\n",
       "      <td>RNA</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>UNC-LCCC</td>\n",
       "      <td>...</td>\n",
       "      <td>13daa1a0-a236-474e-b621-eb131be34af1</td>\n",
       "      <td>3768cf73-e389-449a-bde3-29eb20c9e249</td>\n",
       "      <td>f0f4aa30-1bee-42a9-b109-75c60a42da76</td>\n",
       "      <td>56</td>\n",
       "      <td></td>\n",
       "      <td>2013-09-23</td>\n",
       "      <td>2013-09-23</td>\n",
       "      <td>2013-09-23</td>\n",
       "      <td>Live</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   study                       barcode disease  \\\n",
       "3   TCGA  TCGA-DQ-5630-01A-01R-1873-07    HNSC   \n",
       "8   TCGA  TCGA-KK-A7B0-01A-11R-A32O-07    PRAD   \n",
       "10  TCGA  TCGA-DD-A3A3-11A-11R-A22L-07    LIHC   \n",
       "13  TCGA  TCGA-VV-A829-01A-21R-A36H-07     LGG   \n",
       "15  TCGA  TCGA-56-7222-01A-11R-2045-07    LUSC   \n",
       "\n",
       "                             disease_name sample_type sample_type_name  \\\n",
       "3   Head and Neck squamous cell carcinoma          TP               01   \n",
       "8                 Prostate adenocarcinoma          TP               01   \n",
       "10         Liver hepatocellular carcinoma          NT               11   \n",
       "13               Brain Lower Grade Glioma          TP               01   \n",
       "15           Lung squamous cell carcinoma          TP               01   \n",
       "\n",
       "   analyte_type library_type    center center_name  ...  \\\n",
       "3           RNA      RNA-Seq  UNC-LCCC    UNC-LCCC  ...   \n",
       "8           RNA      RNA-Seq  UNC-LCCC    UNC-LCCC  ...   \n",
       "10          RNA      RNA-Seq  UNC-LCCC    UNC-LCCC  ...   \n",
       "13          RNA      RNA-Seq  UNC-LCCC    UNC-LCCC  ...   \n",
       "15          RNA      RNA-Seq  UNC-LCCC    UNC-LCCC  ...   \n",
       "\n",
       "                              aliquot_id  \\\n",
       "3   13c839d1-f77b-4bde-9bd5-8d5165d94346   \n",
       "8   13d2199f-d552-4ea3-9d7c-c2d1fb01d7a7   \n",
       "10  13d4c229-cf05-4fca-a52d-798dc7894ba1   \n",
       "13  13d82ac1-eecc-41bb-af05-4fd4f614f260   \n",
       "15  13daa1a0-a236-474e-b621-eb131be34af1   \n",
       "\n",
       "                          participant_id  \\\n",
       "3   91a712b5-a724-48d0-ba06-4f6857683463   \n",
       "8   73d4e806-2c15-4c9e-a978-106fb9494e82   \n",
       "10  516c8594-2973-41e5-a5ce-76d177edfa48   \n",
       "13  10887e42-444d-49fc-aee5-dd0e9fc4ea54   \n",
       "15  3768cf73-e389-449a-bde3-29eb20c9e249   \n",
       "\n",
       "                               sample_id tss_id  sample_accession   published  \\\n",
       "3   a0cdb208-e66f-4ad7-86fa-5077145a4a68     DQ                    2013-09-27   \n",
       "8   e93cc6cc-ce3b-45eb-8928-d5b13d7194af     KK                    2014-01-17   \n",
       "10  1079b0a9-f74e-4d0a-bed0-22aad6ea64e2     DD                    2013-09-21   \n",
       "13  9f00d856-b402-46d2-9bb7-2e9e4c2ca19c     VV                    2014-06-24   \n",
       "15  f0f4aa30-1bee-42a9-b109-75c60a42da76     56                    2013-09-23   \n",
       "\n",
       "      uploaded    modified state reason  \n",
       "3   2013-09-25  2013-09-27  Live         \n",
       "8   2014-01-17  2014-01-17  Live         \n",
       "10  2013-09-21  2013-09-21  Live         \n",
       "13  2014-06-20  2014-06-24  Live         \n",
       "15  2013-09-23  2013-09-23  Live         \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# select rows with our analysis ids\n",
    "all_info_final = all_info[all_info['analysis_id'].isin(all_ids_final)]\n",
    "print(len(all_info_final.index))\n",
    "all_info_final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_info_final.to_csv(\"./tcga_ccle_samples.barcode.with_targeted_analysis_ids.txt\", \n",
    "                      sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step4. matched the curated analysis ID to the TCGA clinical data (Kernel: R)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Updating HTML index of packages in '.Library'\n",
      "Making 'packages.html' ... done\n",
      "Bioconductor version 3.10 (BiocManager 1.30.10), R 3.6.2 (2019-12-12)\n",
      "Installing package(s) 'BiocVersion', 'TCGAbiolinks'\n",
      "also installing the dependencies ‘jpeg’, ‘bit’, ‘lambda.r’, ‘futile.options’, ‘ggsci’, ‘cowplot’, ‘ggsignif’, ‘polynom’, ‘exactRankTests’, ‘mvtnorm’, ‘KMsurv’, ‘km.ci’, ‘GenomicAlignments’, ‘hwriter’, ‘latticeExtra’, ‘geneplotter’, ‘Rhtslib’, ‘rjson’, ‘bit64’, ‘blob’, ‘BH’, ‘futile.logger’, ‘snow’, ‘ggpubr’, ‘maxstat’, ‘survMisc’, ‘Biobase’, ‘ShortRead’, ‘DESeq’, ‘aroma.light’, ‘Rsamtools’, ‘Biostrings’, ‘AnnotationDbi’, ‘locfit’, ‘BiocFileCache’, ‘rappdirs’, ‘GetoptLong’, ‘clue’, ‘GlobalOptions’, ‘png’, ‘R.oo’, ‘R.methodsS3’, ‘DelayedArray’, ‘annotate’, ‘ALL’, ‘RSQLite’, ‘rtracklayer’, ‘BiocParallel’, ‘matrixStats’, ‘downloader’, ‘survminer’, ‘XML’, ‘EDASeq’, ‘edgeR’, ‘biomaRt’, ‘ggthemes’, ‘ComplexHeatmap’, ‘R.utils’, ‘SummarizedExperiment’, ‘genefilter’, ‘ConsensusClusterPlus’, ‘doParallel’, ‘GenomicFeatures’, ‘sva’, ‘limma’, ‘matlab’, ‘circlize’, ‘ggrepel’\n",
      "\n",
      "Updating HTML index of packages in '.Library'\n",
      "Making 'packages.html' ... done\n",
      "Old packages: 'uuid'\n"
     ]
    }
   ],
   "source": [
    "# install TCGA biolinks\n",
    "if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "    install.packages(\"BiocManager\")\n",
    "\n",
    "BiocManager::install(\"TCGAbiolinks\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(TCGAbiolinks)\n",
    "library(plyr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 27</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>study</th><th scope=col>barcode</th><th scope=col>disease</th><th scope=col>disease_name</th><th scope=col>sample_type</th><th scope=col>sample_type_name</th><th scope=col>analyte_type</th><th scope=col>library_type</th><th scope=col>center</th><th scope=col>center_name</th><th scope=col>⋯</th><th scope=col>aliquot_id</th><th scope=col>participant_id</th><th scope=col>sample_id</th><th scope=col>tss_id</th><th scope=col>sample_accession</th><th scope=col>published</th><th scope=col>uploaded</th><th scope=col>modified</th><th scope=col>state</th><th scope=col>reason</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TCGA</td><td>TCGA-DQ-5630-01A-01R-1873-07</td><td>HNSC</td><td>Head and Neck squamous cell carcinoma</td><td>TP</td><td> 1</td><td>RNA</td><td>RNA-Seq</td><td>UNC-LCCC</td><td>UNC-LCCC</td><td>⋯</td><td>13c839d1-f77b-4bde-9bd5-8d5165d94346</td><td>91a712b5-a724-48d0-ba06-4f6857683463</td><td>a0cdb208-e66f-4ad7-86fa-5077145a4a68</td><td>DQ</td><td> </td><td>2013-09-27</td><td>2013-09-25</td><td>2013-09-27</td><td>Live</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>TCGA</td><td>TCGA-KK-A7B0-01A-11R-A32O-07</td><td>PRAD</td><td>Prostate adenocarcinoma              </td><td>TP</td><td> 1</td><td>RNA</td><td>RNA-Seq</td><td>UNC-LCCC</td><td>UNC-LCCC</td><td>⋯</td><td>13d2199f-d552-4ea3-9d7c-c2d1fb01d7a7</td><td>73d4e806-2c15-4c9e-a978-106fb9494e82</td><td>e93cc6cc-ce3b-45eb-8928-d5b13d7194af</td><td>KK</td><td> </td><td>2014-01-17</td><td>2014-01-17</td><td>2014-01-17</td><td>Live</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>TCGA</td><td>TCGA-DD-A3A3-11A-11R-A22L-07</td><td>LIHC</td><td>Liver hepatocellular carcinoma       </td><td>NT</td><td>11</td><td>RNA</td><td>RNA-Seq</td><td>UNC-LCCC</td><td>UNC-LCCC</td><td>⋯</td><td>13d4c229-cf05-4fca-a52d-798dc7894ba1</td><td>516c8594-2973-41e5-a5ce-76d177edfa48</td><td>1079b0a9-f74e-4d0a-bed0-22aad6ea64e2</td><td>DD</td><td> </td><td>2013-09-21</td><td>2013-09-21</td><td>2013-09-21</td><td>Live</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>TCGA</td><td>TCGA-VV-A829-01A-21R-A36H-07</td><td>LGG </td><td>Brain Lower Grade Glioma             </td><td>TP</td><td> 1</td><td>RNA</td><td>RNA-Seq</td><td>UNC-LCCC</td><td>UNC-LCCC</td><td>⋯</td><td>13d82ac1-eecc-41bb-af05-4fd4f614f260</td><td>10887e42-444d-49fc-aee5-dd0e9fc4ea54</td><td>9f00d856-b402-46d2-9bb7-2e9e4c2ca19c</td><td>VV</td><td> </td><td>2014-06-24</td><td>2014-06-20</td><td>2014-06-24</td><td>Live</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>TCGA</td><td>TCGA-56-7222-01A-11R-2045-07</td><td>LUSC</td><td>Lung squamous cell carcinoma         </td><td>TP</td><td> 1</td><td>RNA</td><td>RNA-Seq</td><td>UNC-LCCC</td><td>UNC-LCCC</td><td>⋯</td><td>13daa1a0-a236-474e-b621-eb131be34af1</td><td>3768cf73-e389-449a-bde3-29eb20c9e249</td><td>f0f4aa30-1bee-42a9-b109-75c60a42da76</td><td>56</td><td> </td><td>2013-09-23</td><td>2013-09-23</td><td>2013-09-23</td><td>Live</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>TCGA</td><td>TCGA-FW-A3I3-06A-11R-A21D-07</td><td>SKCM</td><td>Skin Cutaneous Melanoma              </td><td>TM</td><td> 6</td><td>RNA</td><td>RNA-Seq</td><td>UNC-LCCC</td><td>UNC-LCCC</td><td>⋯</td><td>13dd0afe-2ac9-4a9a-a7f1-a807dc33a787</td><td>436a0116-eada-4b8f-aafe-29eb7f3241c4</td><td>4ed01a45-bf48-4a36-9dba-b0866a0c008f</td><td>FW</td><td> </td><td>2013-09-22</td><td>2013-09-22</td><td>2013-09-22</td><td>Live</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 27\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & study & barcode & disease & disease\\_name & sample\\_type & sample\\_type\\_name & analyte\\_type & library\\_type & center & center\\_name & ⋯ & aliquot\\_id & participant\\_id & sample\\_id & tss\\_id & sample\\_accession & published & uploaded & modified & state & reason\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <int> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & TCGA & TCGA-DQ-5630-01A-01R-1873-07 & HNSC & Head and Neck squamous cell carcinoma & TP &  1 & RNA & RNA-Seq & UNC-LCCC & UNC-LCCC & ⋯ & 13c839d1-f77b-4bde-9bd5-8d5165d94346 & 91a712b5-a724-48d0-ba06-4f6857683463 & a0cdb208-e66f-4ad7-86fa-5077145a4a68 & DQ &   & 2013-09-27 & 2013-09-25 & 2013-09-27 & Live & NA\\\\\n",
       "\t2 & TCGA & TCGA-KK-A7B0-01A-11R-A32O-07 & PRAD & Prostate adenocarcinoma               & TP &  1 & RNA & RNA-Seq & UNC-LCCC & UNC-LCCC & ⋯ & 13d2199f-d552-4ea3-9d7c-c2d1fb01d7a7 & 73d4e806-2c15-4c9e-a978-106fb9494e82 & e93cc6cc-ce3b-45eb-8928-d5b13d7194af & KK &   & 2014-01-17 & 2014-01-17 & 2014-01-17 & Live & NA\\\\\n",
       "\t3 & TCGA & TCGA-DD-A3A3-11A-11R-A22L-07 & LIHC & Liver hepatocellular carcinoma        & NT & 11 & RNA & RNA-Seq & UNC-LCCC & UNC-LCCC & ⋯ & 13d4c229-cf05-4fca-a52d-798dc7894ba1 & 516c8594-2973-41e5-a5ce-76d177edfa48 & 1079b0a9-f74e-4d0a-bed0-22aad6ea64e2 & DD &   & 2013-09-21 & 2013-09-21 & 2013-09-21 & Live & NA\\\\\n",
       "\t4 & TCGA & TCGA-VV-A829-01A-21R-A36H-07 & LGG  & Brain Lower Grade Glioma              & TP &  1 & RNA & RNA-Seq & UNC-LCCC & UNC-LCCC & ⋯ & 13d82ac1-eecc-41bb-af05-4fd4f614f260 & 10887e42-444d-49fc-aee5-dd0e9fc4ea54 & 9f00d856-b402-46d2-9bb7-2e9e4c2ca19c & VV &   & 2014-06-24 & 2014-06-20 & 2014-06-24 & Live & NA\\\\\n",
       "\t5 & TCGA & TCGA-56-7222-01A-11R-2045-07 & LUSC & Lung squamous cell carcinoma          & TP &  1 & RNA & RNA-Seq & UNC-LCCC & UNC-LCCC & ⋯ & 13daa1a0-a236-474e-b621-eb131be34af1 & 3768cf73-e389-449a-bde3-29eb20c9e249 & f0f4aa30-1bee-42a9-b109-75c60a42da76 & 56 &   & 2013-09-23 & 2013-09-23 & 2013-09-23 & Live & NA\\\\\n",
       "\t6 & TCGA & TCGA-FW-A3I3-06A-11R-A21D-07 & SKCM & Skin Cutaneous Melanoma               & TM &  6 & RNA & RNA-Seq & UNC-LCCC & UNC-LCCC & ⋯ & 13dd0afe-2ac9-4a9a-a7f1-a807dc33a787 & 436a0116-eada-4b8f-aafe-29eb7f3241c4 & 4ed01a45-bf48-4a36-9dba-b0866a0c008f & FW &   & 2013-09-22 & 2013-09-22 & 2013-09-22 & Live & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 27\n",
       "\n",
       "| <!--/--> | study &lt;fct&gt; | barcode &lt;fct&gt; | disease &lt;fct&gt; | disease_name &lt;fct&gt; | sample_type &lt;fct&gt; | sample_type_name &lt;int&gt; | analyte_type &lt;fct&gt; | library_type &lt;fct&gt; | center &lt;fct&gt; | center_name &lt;fct&gt; | ⋯ ⋯ | aliquot_id &lt;fct&gt; | participant_id &lt;fct&gt; | sample_id &lt;fct&gt; | tss_id &lt;fct&gt; | sample_accession &lt;fct&gt; | published &lt;fct&gt; | uploaded &lt;fct&gt; | modified &lt;fct&gt; | state &lt;fct&gt; | reason &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | TCGA | TCGA-DQ-5630-01A-01R-1873-07 | HNSC | Head and Neck squamous cell carcinoma | TP |  1 | RNA | RNA-Seq | UNC-LCCC | UNC-LCCC | ⋯ | 13c839d1-f77b-4bde-9bd5-8d5165d94346 | 91a712b5-a724-48d0-ba06-4f6857683463 | a0cdb208-e66f-4ad7-86fa-5077145a4a68 | DQ | <!----> | 2013-09-27 | 2013-09-25 | 2013-09-27 | Live | NA |\n",
       "| 2 | TCGA | TCGA-KK-A7B0-01A-11R-A32O-07 | PRAD | Prostate adenocarcinoma               | TP |  1 | RNA | RNA-Seq | UNC-LCCC | UNC-LCCC | ⋯ | 13d2199f-d552-4ea3-9d7c-c2d1fb01d7a7 | 73d4e806-2c15-4c9e-a978-106fb9494e82 | e93cc6cc-ce3b-45eb-8928-d5b13d7194af | KK | <!----> | 2014-01-17 | 2014-01-17 | 2014-01-17 | Live | NA |\n",
       "| 3 | TCGA | TCGA-DD-A3A3-11A-11R-A22L-07 | LIHC | Liver hepatocellular carcinoma        | NT | 11 | RNA | RNA-Seq | UNC-LCCC | UNC-LCCC | ⋯ | 13d4c229-cf05-4fca-a52d-798dc7894ba1 | 516c8594-2973-41e5-a5ce-76d177edfa48 | 1079b0a9-f74e-4d0a-bed0-22aad6ea64e2 | DD | <!----> | 2013-09-21 | 2013-09-21 | 2013-09-21 | Live | NA |\n",
       "| 4 | TCGA | TCGA-VV-A829-01A-21R-A36H-07 | LGG  | Brain Lower Grade Glioma              | TP |  1 | RNA | RNA-Seq | UNC-LCCC | UNC-LCCC | ⋯ | 13d82ac1-eecc-41bb-af05-4fd4f614f260 | 10887e42-444d-49fc-aee5-dd0e9fc4ea54 | 9f00d856-b402-46d2-9bb7-2e9e4c2ca19c | VV | <!----> | 2014-06-24 | 2014-06-20 | 2014-06-24 | Live | NA |\n",
       "| 5 | TCGA | TCGA-56-7222-01A-11R-2045-07 | LUSC | Lung squamous cell carcinoma          | TP |  1 | RNA | RNA-Seq | UNC-LCCC | UNC-LCCC | ⋯ | 13daa1a0-a236-474e-b621-eb131be34af1 | 3768cf73-e389-449a-bde3-29eb20c9e249 | f0f4aa30-1bee-42a9-b109-75c60a42da76 | 56 | <!----> | 2013-09-23 | 2013-09-23 | 2013-09-23 | Live | NA |\n",
       "| 6 | TCGA | TCGA-FW-A3I3-06A-11R-A21D-07 | SKCM | Skin Cutaneous Melanoma               | TM |  6 | RNA | RNA-Seq | UNC-LCCC | UNC-LCCC | ⋯ | 13dd0afe-2ac9-4a9a-a7f1-a807dc33a787 | 436a0116-eada-4b8f-aafe-29eb7f3241c4 | 4ed01a45-bf48-4a36-9dba-b0866a0c008f | FW | <!----> | 2013-09-22 | 2013-09-22 | 2013-09-22 | Live | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  study barcode                      disease\n",
       "1 TCGA  TCGA-DQ-5630-01A-01R-1873-07 HNSC   \n",
       "2 TCGA  TCGA-KK-A7B0-01A-11R-A32O-07 PRAD   \n",
       "3 TCGA  TCGA-DD-A3A3-11A-11R-A22L-07 LIHC   \n",
       "4 TCGA  TCGA-VV-A829-01A-21R-A36H-07 LGG    \n",
       "5 TCGA  TCGA-56-7222-01A-11R-2045-07 LUSC   \n",
       "6 TCGA  TCGA-FW-A3I3-06A-11R-A21D-07 SKCM   \n",
       "  disease_name                          sample_type sample_type_name\n",
       "1 Head and Neck squamous cell carcinoma TP           1              \n",
       "2 Prostate adenocarcinoma               TP           1              \n",
       "3 Liver hepatocellular carcinoma        NT          11              \n",
       "4 Brain Lower Grade Glioma              TP           1              \n",
       "5 Lung squamous cell carcinoma          TP           1              \n",
       "6 Skin Cutaneous Melanoma               TM           6              \n",
       "  analyte_type library_type center   center_name ⋯\n",
       "1 RNA          RNA-Seq      UNC-LCCC UNC-LCCC    ⋯\n",
       "2 RNA          RNA-Seq      UNC-LCCC UNC-LCCC    ⋯\n",
       "3 RNA          RNA-Seq      UNC-LCCC UNC-LCCC    ⋯\n",
       "4 RNA          RNA-Seq      UNC-LCCC UNC-LCCC    ⋯\n",
       "5 RNA          RNA-Seq      UNC-LCCC UNC-LCCC    ⋯\n",
       "6 RNA          RNA-Seq      UNC-LCCC UNC-LCCC    ⋯\n",
       "  aliquot_id                           participant_id                      \n",
       "1 13c839d1-f77b-4bde-9bd5-8d5165d94346 91a712b5-a724-48d0-ba06-4f6857683463\n",
       "2 13d2199f-d552-4ea3-9d7c-c2d1fb01d7a7 73d4e806-2c15-4c9e-a978-106fb9494e82\n",
       "3 13d4c229-cf05-4fca-a52d-798dc7894ba1 516c8594-2973-41e5-a5ce-76d177edfa48\n",
       "4 13d82ac1-eecc-41bb-af05-4fd4f614f260 10887e42-444d-49fc-aee5-dd0e9fc4ea54\n",
       "5 13daa1a0-a236-474e-b621-eb131be34af1 3768cf73-e389-449a-bde3-29eb20c9e249\n",
       "6 13dd0afe-2ac9-4a9a-a7f1-a807dc33a787 436a0116-eada-4b8f-aafe-29eb7f3241c4\n",
       "  sample_id                            tss_id sample_accession published \n",
       "1 a0cdb208-e66f-4ad7-86fa-5077145a4a68 DQ                      2013-09-27\n",
       "2 e93cc6cc-ce3b-45eb-8928-d5b13d7194af KK                      2014-01-17\n",
       "3 1079b0a9-f74e-4d0a-bed0-22aad6ea64e2 DD                      2013-09-21\n",
       "4 9f00d856-b402-46d2-9bb7-2e9e4c2ca19c VV                      2014-06-24\n",
       "5 f0f4aa30-1bee-42a9-b109-75c60a42da76 56                      2013-09-23\n",
       "6 4ed01a45-bf48-4a36-9dba-b0866a0c008f FW                      2013-09-22\n",
       "  uploaded   modified   state reason\n",
       "1 2013-09-25 2013-09-27 Live  NA    \n",
       "2 2014-01-17 2014-01-17 Live  NA    \n",
       "3 2013-09-21 2013-09-21 Live  NA    \n",
       "4 2014-06-20 2014-06-24 Live  NA    \n",
       "5 2013-09-23 2013-09-23 Live  NA    \n",
       "6 2013-09-22 2013-09-22 Live  NA    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# read the curated analysis ID from step3\n",
    "data_id <- read.table(\"./tcga_ccle_samples.barcode.with_targeted_analysis_ids.txt\", \n",
    "                     sep = \"\\t\", header=TRUE)\n",
    "head(data_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'TCGA-DQ-5630'</li><li>'TCGA-KK-A7B0'</li><li>'TCGA-DD-A3A3'</li><li>'TCGA-VV-A829'</li><li>'TCGA-56-7222'</li><li>'TCGA-FW-A3I3'</li><li>'TCGA-55-6972'</li><li>'TCGA-H7-A6C4'</li><li>'TCGA-CW-5584'</li><li>'TCGA-KN-8433'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'TCGA-DQ-5630'\n",
       "\\item 'TCGA-KK-A7B0'\n",
       "\\item 'TCGA-DD-A3A3'\n",
       "\\item 'TCGA-VV-A829'\n",
       "\\item 'TCGA-56-7222'\n",
       "\\item 'TCGA-FW-A3I3'\n",
       "\\item 'TCGA-55-6972'\n",
       "\\item 'TCGA-H7-A6C4'\n",
       "\\item 'TCGA-CW-5584'\n",
       "\\item 'TCGA-KN-8433'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'TCGA-DQ-5630'\n",
       "2. 'TCGA-KK-A7B0'\n",
       "3. 'TCGA-DD-A3A3'\n",
       "4. 'TCGA-VV-A829'\n",
       "5. 'TCGA-56-7222'\n",
       "6. 'TCGA-FW-A3I3'\n",
       "7. 'TCGA-55-6972'\n",
       "8. 'TCGA-H7-A6C4'\n",
       "9. 'TCGA-CW-5584'\n",
       "10. 'TCGA-KN-8433'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"TCGA-DQ-5630\" \"TCGA-KK-A7B0\" \"TCGA-DD-A3A3\" \"TCGA-VV-A829\" \"TCGA-56-7222\"\n",
       " [6] \"TCGA-FW-A3I3\" \"TCGA-55-6972\" \"TCGA-H7-A6C4\" \"TCGA-CW-5584\" \"TCGA-KN-8433\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# get the first three elements in the existing barcodes for case barcodes\n",
    "data_id$barcode_final <- unlist(lapply(as.character(data_id$barcode), \n",
    "                                       function(x) paste(unlist(strsplit(x, split = \"-\"))[1:3], collapse = \"-\")  ))\n",
    "                                       \n",
    "data_id$barcode_final[1:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'TCGA-HNSC'</li><li>'TCGA-PRAD'</li><li>'TCGA-LIHC'</li><li>'TCGA-LGG'</li><li>'TCGA-LUSC'</li><li>'TCGA-SKCM'</li><li>'TCGA-LUAD'</li><li>'TCGA-KIRC'</li><li>'TCGA-KICH'</li><li>'TCGA-UCEC'</li><li>'TCGA-PCPG'</li><li>'TCGA-KIRP'</li><li>'TCGA-STAD'</li><li>'TCGA-OV'</li><li>'TCGA-CESC'</li><li>'TCGA-THCA'</li><li>'TCGA-BRCA'</li><li>'TCGA-SARC'</li><li>'TCGA-DLBC'</li><li>'TCGA-ESCA'</li><li>'TCGA-MESO'</li><li>'TCGA-THYM'</li><li>'TCGA-COAD'</li><li>'TCGA-LAML'</li><li>'TCGA-GBM'</li><li>'TCGA-PAAD'</li><li>'TCGA-TGCT'</li><li>'TCGA-READ'</li><li>'TCGA-BLCA'</li><li>'TCGA-UCS'</li><li>'TCGA-UVM'</li><li>'TCGA-CHOL'</li><li>'TCGA-ACC'</li><li>'TCGA-CNTL'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'TCGA-HNSC'\n",
       "\\item 'TCGA-PRAD'\n",
       "\\item 'TCGA-LIHC'\n",
       "\\item 'TCGA-LGG'\n",
       "\\item 'TCGA-LUSC'\n",
       "\\item 'TCGA-SKCM'\n",
       "\\item 'TCGA-LUAD'\n",
       "\\item 'TCGA-KIRC'\n",
       "\\item 'TCGA-KICH'\n",
       "\\item 'TCGA-UCEC'\n",
       "\\item 'TCGA-PCPG'\n",
       "\\item 'TCGA-KIRP'\n",
       "\\item 'TCGA-STAD'\n",
       "\\item 'TCGA-OV'\n",
       "\\item 'TCGA-CESC'\n",
       "\\item 'TCGA-THCA'\n",
       "\\item 'TCGA-BRCA'\n",
       "\\item 'TCGA-SARC'\n",
       "\\item 'TCGA-DLBC'\n",
       "\\item 'TCGA-ESCA'\n",
       "\\item 'TCGA-MESO'\n",
       "\\item 'TCGA-THYM'\n",
       "\\item 'TCGA-COAD'\n",
       "\\item 'TCGA-LAML'\n",
       "\\item 'TCGA-GBM'\n",
       "\\item 'TCGA-PAAD'\n",
       "\\item 'TCGA-TGCT'\n",
       "\\item 'TCGA-READ'\n",
       "\\item 'TCGA-BLCA'\n",
       "\\item 'TCGA-UCS'\n",
       "\\item 'TCGA-UVM'\n",
       "\\item 'TCGA-CHOL'\n",
       "\\item 'TCGA-ACC'\n",
       "\\item 'TCGA-CNTL'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'TCGA-HNSC'\n",
       "2. 'TCGA-PRAD'\n",
       "3. 'TCGA-LIHC'\n",
       "4. 'TCGA-LGG'\n",
       "5. 'TCGA-LUSC'\n",
       "6. 'TCGA-SKCM'\n",
       "7. 'TCGA-LUAD'\n",
       "8. 'TCGA-KIRC'\n",
       "9. 'TCGA-KICH'\n",
       "10. 'TCGA-UCEC'\n",
       "11. 'TCGA-PCPG'\n",
       "12. 'TCGA-KIRP'\n",
       "13. 'TCGA-STAD'\n",
       "14. 'TCGA-OV'\n",
       "15. 'TCGA-CESC'\n",
       "16. 'TCGA-THCA'\n",
       "17. 'TCGA-BRCA'\n",
       "18. 'TCGA-SARC'\n",
       "19. 'TCGA-DLBC'\n",
       "20. 'TCGA-ESCA'\n",
       "21. 'TCGA-MESO'\n",
       "22. 'TCGA-THYM'\n",
       "23. 'TCGA-COAD'\n",
       "24. 'TCGA-LAML'\n",
       "25. 'TCGA-GBM'\n",
       "26. 'TCGA-PAAD'\n",
       "27. 'TCGA-TGCT'\n",
       "28. 'TCGA-READ'\n",
       "29. 'TCGA-BLCA'\n",
       "30. 'TCGA-UCS'\n",
       "31. 'TCGA-UVM'\n",
       "32. 'TCGA-CHOL'\n",
       "33. 'TCGA-ACC'\n",
       "34. 'TCGA-CNTL'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"TCGA-HNSC\" \"TCGA-PRAD\" \"TCGA-LIHC\" \"TCGA-LGG\"  \"TCGA-LUSC\" \"TCGA-SKCM\"\n",
       " [7] \"TCGA-LUAD\" \"TCGA-KIRC\" \"TCGA-KICH\" \"TCGA-UCEC\" \"TCGA-PCPG\" \"TCGA-KIRP\"\n",
       "[13] \"TCGA-STAD\" \"TCGA-OV\"   \"TCGA-CESC\" \"TCGA-THCA\" \"TCGA-BRCA\" \"TCGA-SARC\"\n",
       "[19] \"TCGA-DLBC\" \"TCGA-ESCA\" \"TCGA-MESO\" \"TCGA-THYM\" \"TCGA-COAD\" \"TCGA-LAML\"\n",
       "[25] \"TCGA-GBM\"  \"TCGA-PAAD\" \"TCGA-TGCT\" \"TCGA-READ\" \"TCGA-BLCA\" \"TCGA-UCS\" \n",
       "[31] \"TCGA-UVM\"  \"TCGA-CHOL\" \"TCGA-ACC\"  \"TCGA-CNTL\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_id$project_id <- paste(data_id$study, data_id$disease, sep = \"-\")\n",
    "data_id_uniq = unique(data_id$project_id)\n",
    "data_id_uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'TCGA-HNSC'</li><li>'TCGA-PRAD'</li><li>'TCGA-LIHC'</li><li>'TCGA-LGG'</li><li>'TCGA-LUSC'</li><li>'TCGA-SKCM'</li><li>'TCGA-LUAD'</li><li>'TCGA-KIRC'</li><li>'TCGA-KICH'</li><li>'TCGA-UCEC'</li><li>'TCGA-PCPG'</li><li>'TCGA-KIRP'</li><li>'TCGA-STAD'</li><li>'TCGA-OV'</li><li>'TCGA-CESC'</li><li>'TCGA-THCA'</li><li>'TCGA-BRCA'</li><li>'TCGA-SARC'</li><li>'TCGA-DLBC'</li><li>'TCGA-ESCA'</li><li>'TCGA-MESO'</li><li>'TCGA-THYM'</li><li>'TCGA-COAD'</li><li>'TCGA-LAML'</li><li>'TCGA-GBM'</li><li>'TCGA-PAAD'</li><li>'TCGA-TGCT'</li><li>'TCGA-READ'</li><li>'TCGA-BLCA'</li><li>'TCGA-UCS'</li><li>'TCGA-UVM'</li><li>'TCGA-CHOL'</li><li>'TCGA-ACC'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'TCGA-HNSC'\n",
       "\\item 'TCGA-PRAD'\n",
       "\\item 'TCGA-LIHC'\n",
       "\\item 'TCGA-LGG'\n",
       "\\item 'TCGA-LUSC'\n",
       "\\item 'TCGA-SKCM'\n",
       "\\item 'TCGA-LUAD'\n",
       "\\item 'TCGA-KIRC'\n",
       "\\item 'TCGA-KICH'\n",
       "\\item 'TCGA-UCEC'\n",
       "\\item 'TCGA-PCPG'\n",
       "\\item 'TCGA-KIRP'\n",
       "\\item 'TCGA-STAD'\n",
       "\\item 'TCGA-OV'\n",
       "\\item 'TCGA-CESC'\n",
       "\\item 'TCGA-THCA'\n",
       "\\item 'TCGA-BRCA'\n",
       "\\item 'TCGA-SARC'\n",
       "\\item 'TCGA-DLBC'\n",
       "\\item 'TCGA-ESCA'\n",
       "\\item 'TCGA-MESO'\n",
       "\\item 'TCGA-THYM'\n",
       "\\item 'TCGA-COAD'\n",
       "\\item 'TCGA-LAML'\n",
       "\\item 'TCGA-GBM'\n",
       "\\item 'TCGA-PAAD'\n",
       "\\item 'TCGA-TGCT'\n",
       "\\item 'TCGA-READ'\n",
       "\\item 'TCGA-BLCA'\n",
       "\\item 'TCGA-UCS'\n",
       "\\item 'TCGA-UVM'\n",
       "\\item 'TCGA-CHOL'\n",
       "\\item 'TCGA-ACC'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'TCGA-HNSC'\n",
       "2. 'TCGA-PRAD'\n",
       "3. 'TCGA-LIHC'\n",
       "4. 'TCGA-LGG'\n",
       "5. 'TCGA-LUSC'\n",
       "6. 'TCGA-SKCM'\n",
       "7. 'TCGA-LUAD'\n",
       "8. 'TCGA-KIRC'\n",
       "9. 'TCGA-KICH'\n",
       "10. 'TCGA-UCEC'\n",
       "11. 'TCGA-PCPG'\n",
       "12. 'TCGA-KIRP'\n",
       "13. 'TCGA-STAD'\n",
       "14. 'TCGA-OV'\n",
       "15. 'TCGA-CESC'\n",
       "16. 'TCGA-THCA'\n",
       "17. 'TCGA-BRCA'\n",
       "18. 'TCGA-SARC'\n",
       "19. 'TCGA-DLBC'\n",
       "20. 'TCGA-ESCA'\n",
       "21. 'TCGA-MESO'\n",
       "22. 'TCGA-THYM'\n",
       "23. 'TCGA-COAD'\n",
       "24. 'TCGA-LAML'\n",
       "25. 'TCGA-GBM'\n",
       "26. 'TCGA-PAAD'\n",
       "27. 'TCGA-TGCT'\n",
       "28. 'TCGA-READ'\n",
       "29. 'TCGA-BLCA'\n",
       "30. 'TCGA-UCS'\n",
       "31. 'TCGA-UVM'\n",
       "32. 'TCGA-CHOL'\n",
       "33. 'TCGA-ACC'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"TCGA-HNSC\" \"TCGA-PRAD\" \"TCGA-LIHC\" \"TCGA-LGG\"  \"TCGA-LUSC\" \"TCGA-SKCM\"\n",
       " [7] \"TCGA-LUAD\" \"TCGA-KIRC\" \"TCGA-KICH\" \"TCGA-UCEC\" \"TCGA-PCPG\" \"TCGA-KIRP\"\n",
       "[13] \"TCGA-STAD\" \"TCGA-OV\"   \"TCGA-CESC\" \"TCGA-THCA\" \"TCGA-BRCA\" \"TCGA-SARC\"\n",
       "[19] \"TCGA-DLBC\" \"TCGA-ESCA\" \"TCGA-MESO\" \"TCGA-THYM\" \"TCGA-COAD\" \"TCGA-LAML\"\n",
       "[25] \"TCGA-GBM\"  \"TCGA-PAAD\" \"TCGA-TGCT\" \"TCGA-READ\" \"TCGA-BLCA\" \"TCGA-UCS\" \n",
       "[31] \"TCGA-UVM\"  \"TCGA-CHOL\" \"TCGA-ACC\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# No data clinical data for TCGA-CNTL and only one RNA-seq sample for this cancer type. so remove here\n",
    "data_id_uniq_filter = data_id_uniq[data_id_uniq != 'TCGA-CNTL']\n",
    "data_id_uniq_filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"TCGA-HNSC\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-PRAD\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-LIHC\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-LGG\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-LUSC\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-SKCM\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-LUAD\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-KIRC\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-KICH\"\n",
      "  |===================================================================== |  99%\n",
      "  |===================================================================== |  99%\n",
      "  |===================================================================== |  99%\n",
      "[1] \"TCGA-UCEC\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================|  99%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-PCPG\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-KIRP\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-STAD\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-OV\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-CESC\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-THCA\"\n",
      "  |======================================================================| 100%\n",
      "  |===================================================================== |  99%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-BRCA\"\n",
      "  |======================================================================| 100%\n",
      "  |===================================================================== |  98%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-SARC\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-DLBC\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-ESCA\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-MESO\"\n",
      "  |======================================================================| 100%\n",
      "  |==================================================================    |  94%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-THYM\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-COAD\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-LAML\"\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-GBM\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-PAAD\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-TGCT\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-READ\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================|  99%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-BLCA\"\n",
      "  |======================================================================| 100%\n",
      "  |===================================================================   |  95%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-UCS\"\n",
      "  |======================================================================| 100%\n",
      "  |===================================================================== |  98%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-UVM\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-CHOL\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "[1] \"TCGA-ACC\"\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n"
     ]
    }
   ],
   "source": [
    "# get all clinical data for each project id and compile them\n",
    "# Since it dynamically links to TCGA server, be careful of internet connection\n",
    "clinical_all <- data.frame()\n",
    "for (pro_id in data_id_uniq_filter){\n",
    "    print(pro_id)\n",
    "    barcode_sub <- data_id[which(data_id$project_id == pro_id),\"barcode_final\"]\n",
    "    suppressMessages(query_id <- GDCquery(barcode = barcode_sub, file.type = \"xml\",\n",
    "                 project = pro_id, data.category=\"Clinical\"))\n",
    "    \n",
    "    suppressMessages(GDCdownload(query_id))\n",
    "    suppressMessages(clinical <- GDCprepare_clinic(query_id, clinical.info = \"patient\"))\n",
    "    clinical_all <- rbind.fill(clinical_all, clinical)\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>11378</li><li>641</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 11378\n",
       "\\item 641\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 11378\n",
       "2. 641\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 11378   641"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>10357</li><li>641</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10357\n",
       "\\item 641\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10357\n",
       "2. 641\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10357   641"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 641</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>bcr_patient_barcode</th><th scope=col>additional_studies</th><th scope=col>tumor_tissue_site</th><th scope=col>histological_type</th><th scope=col>tissue_prospective_collection_indicator</th><th scope=col>tissue_retrospective_collection_indicator</th><th scope=col>gender</th><th scope=col>vital_status</th><th scope=col>days_to_birth</th><th scope=col>days_to_last_known_alive</th><th scope=col>⋯</th><th scope=col>mitotane_therapy_for_macroscopic_residual_disease</th><th scope=col>therapeutic_mitotane_lvl_macroscopic_residual</th><th scope=col>therapeutic_mitotane_lvl_progression</th><th scope=col>primary_pathology_ct_scan_findings</th><th scope=col>primary_pathology_weiss_assessment_report</th><th scope=col>primary_pathology_mitoses_count</th><th scope=col>primary_pathology_metastatic_neoplasm_confirmed_diagnosis_method_names</th><th scope=col>primary_pathology_metastatic_neoplasm_initial_diagnosis_anatomic_sites</th><th scope=col>primary_pathology_excess_adrenal_hormone_history_types</th><th scope=col>primary_pathology_germline_genotype_testing_report</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TCGA-4P-AA8J</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td><span style=white-space:pre-wrap>MALE  </span></td><td>Alive</td><td>-24222</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>TCGA-BA-4074</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td><span style=white-space:pre-wrap>MALE  </span></td><td>Dead </td><td>-25282</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>TCGA-BA-4075</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td><span style=white-space:pre-wrap>MALE  </span></td><td>Dead </td><td>-17951</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>TCGA-BA-4076</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td><span style=white-space:pre-wrap>MALE  </span></td><td>Dead </td><td>-14405</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>TCGA-BA-4077</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td>FEMALE</td><td>Dead </td><td>-16536</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>TCGA-BA-4078</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td><span style=white-space:pre-wrap>MALE  </span></td><td>Dead </td><td>-30480</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 641\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & bcr\\_patient\\_barcode & additional\\_studies & tumor\\_tissue\\_site & histological\\_type & tissue\\_prospective\\_collection\\_indicator & tissue\\_retrospective\\_collection\\_indicator & gender & vital\\_status & days\\_to\\_birth & days\\_to\\_last\\_known\\_alive & ⋯ & mitotane\\_therapy\\_for\\_macroscopic\\_residual\\_disease & therapeutic\\_mitotane\\_lvl\\_macroscopic\\_residual & therapeutic\\_mitotane\\_lvl\\_progression & primary\\_pathology\\_ct\\_scan\\_findings & primary\\_pathology\\_weiss\\_assessment\\_report & primary\\_pathology\\_mitoses\\_count & primary\\_pathology\\_metastatic\\_neoplasm\\_confirmed\\_diagnosis\\_method\\_names & primary\\_pathology\\_metastatic\\_neoplasm\\_initial\\_diagnosis\\_anatomic\\_sites & primary\\_pathology\\_excess\\_adrenal\\_hormone\\_history\\_types & primary\\_pathology\\_germline\\_genotype\\_testing\\_report\\\\\n",
       "  & <chr> & <int> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & ⋯ & <fct> & <fct> & <fct> & <fct> & <chr> & <int> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & TCGA-4P-AA8J & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & MALE   & Alive & -24222 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t2 & TCGA-BA-4074 & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & MALE   & Dead  & -25282 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t3 & TCGA-BA-4075 & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & MALE   & Dead  & -17951 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t5 & TCGA-BA-4076 & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & MALE   & Dead  & -14405 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t6 & TCGA-BA-4077 & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & FEMALE & Dead  & -16536 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t8 & TCGA-BA-4078 & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & MALE   & Dead  & -30480 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 641\n",
       "\n",
       "| <!--/--> | bcr_patient_barcode &lt;chr&gt; | additional_studies &lt;int&gt; | tumor_tissue_site &lt;fct&gt; | histological_type &lt;fct&gt; | tissue_prospective_collection_indicator &lt;fct&gt; | tissue_retrospective_collection_indicator &lt;fct&gt; | gender &lt;fct&gt; | vital_status &lt;fct&gt; | days_to_birth &lt;int&gt; | days_to_last_known_alive &lt;int&gt; | ⋯ ⋯ | mitotane_therapy_for_macroscopic_residual_disease &lt;fct&gt; | therapeutic_mitotane_lvl_macroscopic_residual &lt;fct&gt; | therapeutic_mitotane_lvl_progression &lt;fct&gt; | primary_pathology_ct_scan_findings &lt;fct&gt; | primary_pathology_weiss_assessment_report &lt;chr&gt; | primary_pathology_mitoses_count &lt;int&gt; | primary_pathology_metastatic_neoplasm_confirmed_diagnosis_method_names &lt;fct&gt; | primary_pathology_metastatic_neoplasm_initial_diagnosis_anatomic_sites &lt;fct&gt; | primary_pathology_excess_adrenal_hormone_history_types &lt;fct&gt; | primary_pathology_germline_genotype_testing_report &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | TCGA-4P-AA8J | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | MALE   | Alive | -24222 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 2 | TCGA-BA-4074 | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | MALE   | Dead  | -25282 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 3 | TCGA-BA-4075 | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | MALE   | Dead  | -17951 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 5 | TCGA-BA-4076 | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | MALE   | Dead  | -14405 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 6 | TCGA-BA-4077 | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | FEMALE | Dead  | -16536 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 8 | TCGA-BA-4078 | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | MALE   | Dead  | -30480 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  bcr_patient_barcode additional_studies tumor_tissue_site\n",
       "1 TCGA-4P-AA8J        NA                 Head and Neck    \n",
       "2 TCGA-BA-4074        NA                 Head and Neck    \n",
       "3 TCGA-BA-4075        NA                 Head and Neck    \n",
       "5 TCGA-BA-4076        NA                 Head and Neck    \n",
       "6 TCGA-BA-4077        NA                 Head and Neck    \n",
       "8 TCGA-BA-4078        NA                 Head and Neck    \n",
       "  histological_type                   tissue_prospective_collection_indicator\n",
       "1 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "2 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "3 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "5 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "6 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "8 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "  tissue_retrospective_collection_indicator gender vital_status days_to_birth\n",
       "1 YES                                       MALE   Alive        -24222       \n",
       "2 YES                                       MALE   Dead         -25282       \n",
       "3 YES                                       MALE   Dead         -17951       \n",
       "5 YES                                       MALE   Dead         -14405       \n",
       "6 YES                                       FEMALE Dead         -16536       \n",
       "8 YES                                       MALE   Dead         -30480       \n",
       "  days_to_last_known_alive ⋯ mitotane_therapy_for_macroscopic_residual_disease\n",
       "1 NA                       ⋯ NA                                               \n",
       "2 NA                       ⋯ NA                                               \n",
       "3 NA                       ⋯ NA                                               \n",
       "5 NA                       ⋯ NA                                               \n",
       "6 NA                       ⋯ NA                                               \n",
       "8 NA                       ⋯ NA                                               \n",
       "  therapeutic_mitotane_lvl_macroscopic_residual\n",
       "1 NA                                           \n",
       "2 NA                                           \n",
       "3 NA                                           \n",
       "5 NA                                           \n",
       "6 NA                                           \n",
       "8 NA                                           \n",
       "  therapeutic_mitotane_lvl_progression primary_pathology_ct_scan_findings\n",
       "1 NA                                   NA                                \n",
       "2 NA                                   NA                                \n",
       "3 NA                                   NA                                \n",
       "5 NA                                   NA                                \n",
       "6 NA                                   NA                                \n",
       "8 NA                                   NA                                \n",
       "  primary_pathology_weiss_assessment_report primary_pathology_mitoses_count\n",
       "1 NA                                        NA                             \n",
       "2 NA                                        NA                             \n",
       "3 NA                                        NA                             \n",
       "5 NA                                        NA                             \n",
       "6 NA                                        NA                             \n",
       "8 NA                                        NA                             \n",
       "  primary_pathology_metastatic_neoplasm_confirmed_diagnosis_method_names\n",
       "1 NA                                                                    \n",
       "2 NA                                                                    \n",
       "3 NA                                                                    \n",
       "5 NA                                                                    \n",
       "6 NA                                                                    \n",
       "8 NA                                                                    \n",
       "  primary_pathology_metastatic_neoplasm_initial_diagnosis_anatomic_sites\n",
       "1 NA                                                                    \n",
       "2 NA                                                                    \n",
       "3 NA                                                                    \n",
       "5 NA                                                                    \n",
       "6 NA                                                                    \n",
       "8 NA                                                                    \n",
       "  primary_pathology_excess_adrenal_hormone_history_types\n",
       "1 NA                                                    \n",
       "2 NA                                                    \n",
       "3 NA                                                    \n",
       "5 NA                                                    \n",
       "6 NA                                                    \n",
       "8 NA                                                    \n",
       "  primary_pathology_germline_genotype_testing_report\n",
       "1 NA                                                \n",
       "2 NA                                                \n",
       "3 NA                                                \n",
       "5 NA                                                \n",
       "6 NA                                                \n",
       "8 NA                                                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(clinical_all)\n",
    "clinical_all_unique <- unique(clinical_all)\n",
    "dim(clinical_all_unique)\n",
    "head(clinical_all_unique)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(clinical_all_unique, \"TCGA_clinical_patient_records_for_each_barcode.txt\",\n",
    "           sep=\"\\t\", col.names=TRUE, row.names=FALSE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get the patient clinical data types and selected those of interest\n",
    "write.table(colnames(clinical_all_unique),\"TCGA_clinical_patient_records_for_each_barcode_colname.txt\",\n",
    "           sep=\"\\t\", row.names=FALSE, col.names=FALSE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 32</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>bcr_patient_barcode</th><th scope=col>additional_studies</th><th scope=col>tumor_tissue_site</th><th scope=col>histological_type</th><th scope=col>tissue_prospective_collection_indicator</th><th scope=col>tissue_retrospective_collection_indicator</th><th scope=col>gender</th><th scope=col>vital_status</th><th scope=col>days_to_birth</th><th scope=col>days_to_last_known_alive</th><th scope=col>⋯</th><th scope=col>diagnosis</th><th scope=col>number_of_lymphnodes_positive</th><th scope=col>diabetes</th><th scope=col>hypertension</th><th scope=col>tumor_type</th><th scope=col>lymphatic_invasion</th><th scope=col>age_began_smoking_in_years</th><th scope=col>treatment</th><th scope=col>history_of_disease</th><th scope=col>country_of_birth</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TCGA-4P-AA8J</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td><span style=white-space:pre-wrap>MALE  </span></td><td>Alive</td><td>-24222</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>TCGA-BA-4074</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td><span style=white-space:pre-wrap>MALE  </span></td><td>Dead </td><td>-25282</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>TCGA-BA-4075</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td><span style=white-space:pre-wrap>MALE  </span></td><td>Dead </td><td>-17951</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>TCGA-BA-4076</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td><span style=white-space:pre-wrap>MALE  </span></td><td>Dead </td><td>-14405</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>TCGA-BA-4077</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td>FEMALE</td><td>Dead </td><td>-16536</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>TCGA-BA-4078</td><td>NA</td><td>Head and Neck</td><td>Head &amp; Neck Squamous Cell Carcinoma</td><td>NO</td><td>YES</td><td><span style=white-space:pre-wrap>MALE  </span></td><td>Dead </td><td>-30480</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 32\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & bcr\\_patient\\_barcode & additional\\_studies & tumor\\_tissue\\_site & histological\\_type & tissue\\_prospective\\_collection\\_indicator & tissue\\_retrospective\\_collection\\_indicator & gender & vital\\_status & days\\_to\\_birth & days\\_to\\_last\\_known\\_alive & ⋯ & diagnosis & number\\_of\\_lymphnodes\\_positive & diabetes & hypertension & tumor\\_type & lymphatic\\_invasion & age\\_began\\_smoking\\_in\\_years & treatment & history\\_of\\_disease & country\\_of\\_birth\\\\\n",
       "  & <chr> & <int> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & ⋯ & <fct> & <int> & <fct> & <fct> & <fct> & <fct> & <int> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & TCGA-4P-AA8J & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & MALE   & Alive & -24222 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t2 & TCGA-BA-4074 & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & MALE   & Dead  & -25282 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t3 & TCGA-BA-4075 & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & MALE   & Dead  & -17951 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t5 & TCGA-BA-4076 & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & MALE   & Dead  & -14405 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t6 & TCGA-BA-4077 & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & FEMALE & Dead  & -16536 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t8 & TCGA-BA-4078 & NA & Head and Neck & Head \\& Neck Squamous Cell Carcinoma & NO & YES & MALE   & Dead  & -30480 & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 32\n",
       "\n",
       "| <!--/--> | bcr_patient_barcode &lt;chr&gt; | additional_studies &lt;int&gt; | tumor_tissue_site &lt;fct&gt; | histological_type &lt;fct&gt; | tissue_prospective_collection_indicator &lt;fct&gt; | tissue_retrospective_collection_indicator &lt;fct&gt; | gender &lt;fct&gt; | vital_status &lt;fct&gt; | days_to_birth &lt;int&gt; | days_to_last_known_alive &lt;int&gt; | ⋯ ⋯ | diagnosis &lt;fct&gt; | number_of_lymphnodes_positive &lt;int&gt; | diabetes &lt;fct&gt; | hypertension &lt;fct&gt; | tumor_type &lt;fct&gt; | lymphatic_invasion &lt;fct&gt; | age_began_smoking_in_years &lt;int&gt; | treatment &lt;fct&gt; | history_of_disease &lt;fct&gt; | country_of_birth &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | TCGA-4P-AA8J | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | MALE   | Alive | -24222 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 2 | TCGA-BA-4074 | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | MALE   | Dead  | -25282 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 3 | TCGA-BA-4075 | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | MALE   | Dead  | -17951 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 5 | TCGA-BA-4076 | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | MALE   | Dead  | -14405 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 6 | TCGA-BA-4077 | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | FEMALE | Dead  | -16536 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 8 | TCGA-BA-4078 | NA | Head and Neck | Head &amp; Neck Squamous Cell Carcinoma | NO | YES | MALE   | Dead  | -30480 | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  bcr_patient_barcode additional_studies tumor_tissue_site\n",
       "1 TCGA-4P-AA8J        NA                 Head and Neck    \n",
       "2 TCGA-BA-4074        NA                 Head and Neck    \n",
       "3 TCGA-BA-4075        NA                 Head and Neck    \n",
       "5 TCGA-BA-4076        NA                 Head and Neck    \n",
       "6 TCGA-BA-4077        NA                 Head and Neck    \n",
       "8 TCGA-BA-4078        NA                 Head and Neck    \n",
       "  histological_type                   tissue_prospective_collection_indicator\n",
       "1 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "2 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "3 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "5 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "6 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "8 Head & Neck Squamous Cell Carcinoma NO                                     \n",
       "  tissue_retrospective_collection_indicator gender vital_status days_to_birth\n",
       "1 YES                                       MALE   Alive        -24222       \n",
       "2 YES                                       MALE   Dead         -25282       \n",
       "3 YES                                       MALE   Dead         -17951       \n",
       "5 YES                                       MALE   Dead         -14405       \n",
       "6 YES                                       FEMALE Dead         -16536       \n",
       "8 YES                                       MALE   Dead         -30480       \n",
       "  days_to_last_known_alive ⋯ diagnosis number_of_lymphnodes_positive diabetes\n",
       "1 NA                       ⋯ NA        NA                            NA      \n",
       "2 NA                       ⋯ NA        NA                            NA      \n",
       "3 NA                       ⋯ NA        NA                            NA      \n",
       "5 NA                       ⋯ NA        NA                            NA      \n",
       "6 NA                       ⋯ NA        NA                            NA      \n",
       "8 NA                       ⋯ NA        NA                            NA      \n",
       "  hypertension tumor_type lymphatic_invasion age_began_smoking_in_years\n",
       "1 NA           NA         NA                 NA                        \n",
       "2 NA           NA         NA                 NA                        \n",
       "3 NA           NA         NA                 NA                        \n",
       "5 NA           NA         NA                 NA                        \n",
       "6 NA           NA         NA                 NA                        \n",
       "8 NA           NA         NA                 NA                        \n",
       "  treatment history_of_disease country_of_birth\n",
       "1 NA        NA                 NA              \n",
       "2 NA        NA                 NA              \n",
       "3 NA        NA                 NA              \n",
       "5 NA        NA                 NA              \n",
       "6 NA        NA                 NA              \n",
       "8 NA        NA                 NA              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# read the selected data types of interest\n",
    "selected_type <- read.table(\"TCGA_clinical_patient_records_for_each_barcode_colname_selected.txt\", header = FALSE)\n",
    "clinical_all_selected <- clinical_all_unique[, as.character(selected_type$V1)]\n",
    "head(clinical_all_selected)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(clinical_all_selected, \"TCGA_clinical_patient_records_for_each_barcode_selected.txt\",\n",
    "           sep=\"\\t\", col.names=TRUE, row.names=FALSE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>11295</li><li>60</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 11295\n",
       "\\item 60\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 11295\n",
       "2. 60\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 11295    60"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# add the collected TCGA patient clinical information to the curated RNA modification data for each patient\n",
    "data_id_with_clinical <- merge(x = data_id, y = clinical_all_selected,\n",
    "                              by.x=\"barcode_final\", by.y=\"bcr_patient_barcode\")\n",
    "dim(data_id_with_clinical)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(data_id_with_clinical, \"tcga_ccle_samples.barcode.with_targeted_analysis_ids.with_clinical_info.txt\",\n",
    "           sep=\"\\t\", row.names=FALSE, col.names=TRUE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 60</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>barcode_final</th><th scope=col>study</th><th scope=col>barcode</th><th scope=col>disease</th><th scope=col>disease_name</th><th scope=col>sample_type</th><th scope=col>sample_type_name</th><th scope=col>analyte_type</th><th scope=col>library_type</th><th scope=col>center</th><th scope=col>⋯</th><th scope=col>diagnosis</th><th scope=col>number_of_lymphnodes_positive</th><th scope=col>diabetes</th><th scope=col>hypertension</th><th scope=col>tumor_type</th><th scope=col>lymphatic_invasion</th><th scope=col>age_began_smoking_in_years</th><th scope=col>treatment</th><th scope=col>history_of_disease</th><th scope=col>country_of_birth</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>TCGA-02-0047</td><td>TCGA</td><td>TCGA-02-0047-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>TP</td><td>1</td><td>RNA</td><td>RNA-Seq</td><td>BI     </td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA </td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>TCGA-02-0055</td><td>TCGA</td><td>TCGA-02-0055-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>TP</td><td>1</td><td>RNA</td><td>RNA-Seq</td><td>BI     </td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA </td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>TCGA-02-2483</td><td>TCGA</td><td>TCGA-02-2483-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>TP</td><td>1</td><td>RNA</td><td>RNA-Seq</td><td>BI     </td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA </td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>TCGA-02-2485</td><td>TCGA</td><td>TCGA-02-2485-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>TP</td><td>1</td><td>RNA</td><td>RNA-Seq</td><td>BI     </td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA </td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>TCGA-02-2486</td><td>TCGA</td><td>TCGA-02-2486-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>TP</td><td>1</td><td>RNA</td><td>RNA-Seq</td><td>BI     </td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA </td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>TCGA-04-1331</td><td>TCGA</td><td>TCGA-04-1331-01A-01R-1569-13</td><td>OV </td><td>Ovarian serous cystadenocarcinoma</td><td>TP</td><td>1</td><td>RNA</td><td>RNA-Seq</td><td>BCCAGSC</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>YES</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 60\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & barcode\\_final & study & barcode & disease & disease\\_name & sample\\_type & sample\\_type\\_name & analyte\\_type & library\\_type & center & ⋯ & diagnosis & number\\_of\\_lymphnodes\\_positive & diabetes & hypertension & tumor\\_type & lymphatic\\_invasion & age\\_began\\_smoking\\_in\\_years & treatment & history\\_of\\_disease & country\\_of\\_birth\\\\\n",
       "  & <chr> & <fct> & <fct> & <fct> & <fct> & <fct> & <int> & <fct> & <fct> & <fct> & ⋯ & <fct> & <int> & <fct> & <fct> & <fct> & <fct> & <int> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & TCGA-02-0047 & TCGA & TCGA-02-0047-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & TP & 1 & RNA & RNA-Seq & BI      & ⋯ & NA & NA & NA & NA & NA & NA  & NA & NA & NA & NA\\\\\n",
       "\t2 & TCGA-02-0055 & TCGA & TCGA-02-0055-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & TP & 1 & RNA & RNA-Seq & BI      & ⋯ & NA & NA & NA & NA & NA & NA  & NA & NA & NA & NA\\\\\n",
       "\t3 & TCGA-02-2483 & TCGA & TCGA-02-2483-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & TP & 1 & RNA & RNA-Seq & BI      & ⋯ & NA & NA & NA & NA & NA & NA  & NA & NA & NA & NA\\\\\n",
       "\t4 & TCGA-02-2485 & TCGA & TCGA-02-2485-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & TP & 1 & RNA & RNA-Seq & BI      & ⋯ & NA & NA & NA & NA & NA & NA  & NA & NA & NA & NA\\\\\n",
       "\t5 & TCGA-02-2486 & TCGA & TCGA-02-2486-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & TP & 1 & RNA & RNA-Seq & BI      & ⋯ & NA & NA & NA & NA & NA & NA  & NA & NA & NA & NA\\\\\n",
       "\t6 & TCGA-04-1331 & TCGA & TCGA-04-1331-01A-01R-1569-13 & OV  & Ovarian serous cystadenocarcinoma & TP & 1 & RNA & RNA-Seq & BCCAGSC & ⋯ & NA & NA & NA & NA & NA & YES & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 60\n",
       "\n",
       "| <!--/--> | barcode_final &lt;chr&gt; | study &lt;fct&gt; | barcode &lt;fct&gt; | disease &lt;fct&gt; | disease_name &lt;fct&gt; | sample_type &lt;fct&gt; | sample_type_name &lt;int&gt; | analyte_type &lt;fct&gt; | library_type &lt;fct&gt; | center &lt;fct&gt; | ⋯ ⋯ | diagnosis &lt;fct&gt; | number_of_lymphnodes_positive &lt;int&gt; | diabetes &lt;fct&gt; | hypertension &lt;fct&gt; | tumor_type &lt;fct&gt; | lymphatic_invasion &lt;fct&gt; | age_began_smoking_in_years &lt;int&gt; | treatment &lt;fct&gt; | history_of_disease &lt;fct&gt; | country_of_birth &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | TCGA-02-0047 | TCGA | TCGA-02-0047-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | TP | 1 | RNA | RNA-Seq | BI      | ⋯ | NA | NA | NA | NA | NA | NA  | NA | NA | NA | NA |\n",
       "| 2 | TCGA-02-0055 | TCGA | TCGA-02-0055-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | TP | 1 | RNA | RNA-Seq | BI      | ⋯ | NA | NA | NA | NA | NA | NA  | NA | NA | NA | NA |\n",
       "| 3 | TCGA-02-2483 | TCGA | TCGA-02-2483-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | TP | 1 | RNA | RNA-Seq | BI      | ⋯ | NA | NA | NA | NA | NA | NA  | NA | NA | NA | NA |\n",
       "| 4 | TCGA-02-2485 | TCGA | TCGA-02-2485-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | TP | 1 | RNA | RNA-Seq | BI      | ⋯ | NA | NA | NA | NA | NA | NA  | NA | NA | NA | NA |\n",
       "| 5 | TCGA-02-2486 | TCGA | TCGA-02-2486-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | TP | 1 | RNA | RNA-Seq | BI      | ⋯ | NA | NA | NA | NA | NA | NA  | NA | NA | NA | NA |\n",
       "| 6 | TCGA-04-1331 | TCGA | TCGA-04-1331-01A-01R-1569-13 | OV  | Ovarian serous cystadenocarcinoma | TP | 1 | RNA | RNA-Seq | BCCAGSC | ⋯ | NA | NA | NA | NA | NA | YES | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  barcode_final study barcode                      disease\n",
       "1 TCGA-02-0047  TCGA  TCGA-02-0047-01A-01R-1849-01 GBM    \n",
       "2 TCGA-02-0055  TCGA  TCGA-02-0055-01A-01R-1849-01 GBM    \n",
       "3 TCGA-02-2483  TCGA  TCGA-02-2483-01A-01R-1849-01 GBM    \n",
       "4 TCGA-02-2485  TCGA  TCGA-02-2485-01A-01R-1849-01 GBM    \n",
       "5 TCGA-02-2486  TCGA  TCGA-02-2486-01A-01R-1849-01 GBM    \n",
       "6 TCGA-04-1331  TCGA  TCGA-04-1331-01A-01R-1569-13 OV     \n",
       "  disease_name                      sample_type sample_type_name analyte_type\n",
       "1 Glioblastoma multiforme           TP          1                RNA         \n",
       "2 Glioblastoma multiforme           TP          1                RNA         \n",
       "3 Glioblastoma multiforme           TP          1                RNA         \n",
       "4 Glioblastoma multiforme           TP          1                RNA         \n",
       "5 Glioblastoma multiforme           TP          1                RNA         \n",
       "6 Ovarian serous cystadenocarcinoma TP          1                RNA         \n",
       "  library_type center  ⋯ diagnosis number_of_lymphnodes_positive diabetes\n",
       "1 RNA-Seq      BI      ⋯ NA        NA                            NA      \n",
       "2 RNA-Seq      BI      ⋯ NA        NA                            NA      \n",
       "3 RNA-Seq      BI      ⋯ NA        NA                            NA      \n",
       "4 RNA-Seq      BI      ⋯ NA        NA                            NA      \n",
       "5 RNA-Seq      BI      ⋯ NA        NA                            NA      \n",
       "6 RNA-Seq      BCCAGSC ⋯ NA        NA                            NA      \n",
       "  hypertension tumor_type lymphatic_invasion age_began_smoking_in_years\n",
       "1 NA           NA         NA                 NA                        \n",
       "2 NA           NA         NA                 NA                        \n",
       "3 NA           NA         NA                 NA                        \n",
       "4 NA           NA         NA                 NA                        \n",
       "5 NA           NA         NA                 NA                        \n",
       "6 NA           NA         YES                NA                        \n",
       "  treatment history_of_disease country_of_birth\n",
       "1 NA        NA                 NA              \n",
       "2 NA        NA                 NA              \n",
       "3 NA        NA                 NA              \n",
       "4 NA        NA                 NA              \n",
       "5 NA        NA                 NA              \n",
       "6 NA        NA                 NA              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_id_with_clinical)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>analysis_id</th><th scope=col>barcode_final</th><th scope=col>project_id</th><th scope=col>barcode</th><th scope=col>disease</th><th scope=col>disease_name</th><th scope=col>tumor_tissue_site</th><th scope=col>histological_type</th><th scope=col>gender</th><th scope=col>days_to_initial_pathologic_diagnosis</th><th scope=col>age_at_initial_pathologic_diagnosis</th><th scope=col>vital_status</th><th scope=col>days_to_birth</th><th scope=col>days_to_death</th><th scope=col>days_to_last_followup</th><th scope=col>patient_death_reason</th><th scope=col>race_list</th><th scope=col>treatment</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>fb47df4a-b9a7-46c6-9e8d-247c00057185</td><td>TCGA-02-0047</td><td>TCGA-GBM</td><td>TCGA-02-0047-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>78</td><td>Dead </td><td>-28759</td><td> 448</td><td> 448</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>e1a99d0d-6b86-484d-8662-23c4378cde92</td><td>TCGA-02-0055</td><td>TCGA-GBM</td><td>TCGA-02-0055-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>FEMALE</td><td>0</td><td>62</td><td>Dead </td><td>-22798</td><td>  76</td><td>  76</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f</td><td>TCGA-02-2483</td><td>TCGA-GBM</td><td>TCGA-02-2483-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>43</td><td>Alive</td><td>-15964</td><td>  NA</td><td> 466</td><td>NA</td><td>ASIAN                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>f752e5d9-00a8-41f1-ae38-2b589f499088</td><td>TCGA-02-2485</td><td>TCGA-GBM</td><td>TCGA-02-2485-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>53</td><td>Alive</td><td>-19494</td><td>  NA</td><td> 470</td><td>NA</td><td>BLACK OR AFRICAN AMERICAN</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>50127941-3e32-46a5-956e-f55fbfc4080f</td><td>TCGA-02-2486</td><td>TCGA-GBM</td><td>TCGA-02-2486-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>64</td><td>Alive</td><td>-23394</td><td>  NA</td><td> 493</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>2e13491b-c707-49e6-852c-ccd1d6252b00</td><td>TCGA-04-1331</td><td>TCGA-OV </td><td>TCGA-04-1331-01A-01R-1569-13</td><td>OV </td><td>Ovarian serous cystadenocarcinoma</td><td>Ovary</td><td>Serous Cystadenocarcinoma      </td><td>FEMALE</td><td>0</td><td>78</td><td>Dead </td><td>-28848</td><td>1336</td><td>1224</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 18\n",
       "\\begin{tabular}{r|llllllllllllllllll}\n",
       "  & analysis\\_id & barcode\\_final & project\\_id & barcode & disease & disease\\_name & tumor\\_tissue\\_site & histological\\_type & gender & days\\_to\\_initial\\_pathologic\\_diagnosis & age\\_at\\_initial\\_pathologic\\_diagnosis & vital\\_status & days\\_to\\_birth & days\\_to\\_death & days\\_to\\_last\\_followup & patient\\_death\\_reason & race\\_list & treatment\\\\\n",
       "  & <fct> & <chr> & <chr> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & <fct> & <int> & <int> & <int> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & fb47df4a-b9a7-46c6-9e8d-247c00057185 & TCGA-02-0047 & TCGA-GBM & TCGA-02-0047-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 78 & Dead  & -28759 &  448 &  448 & NA & WHITE                     & NA\\\\\n",
       "\t2 & e1a99d0d-6b86-484d-8662-23c4378cde92 & TCGA-02-0055 & TCGA-GBM & TCGA-02-0055-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & FEMALE & 0 & 62 & Dead  & -22798 &   76 &   76 & NA & WHITE                     & NA\\\\\n",
       "\t3 & 00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f & TCGA-02-2483 & TCGA-GBM & TCGA-02-2483-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 43 & Alive & -15964 &   NA &  466 & NA & ASIAN                     & NA\\\\\n",
       "\t4 & f752e5d9-00a8-41f1-ae38-2b589f499088 & TCGA-02-2485 & TCGA-GBM & TCGA-02-2485-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 53 & Alive & -19494 &   NA &  470 & NA & BLACK OR AFRICAN AMERICAN & NA\\\\\n",
       "\t5 & 50127941-3e32-46a5-956e-f55fbfc4080f & TCGA-02-2486 & TCGA-GBM & TCGA-02-2486-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 64 & Alive & -23394 &   NA &  493 & NA & WHITE                     & NA\\\\\n",
       "\t6 & 2e13491b-c707-49e6-852c-ccd1d6252b00 & TCGA-04-1331 & TCGA-OV  & TCGA-04-1331-01A-01R-1569-13 & OV  & Ovarian serous cystadenocarcinoma & Ovary & Serous Cystadenocarcinoma       & FEMALE & 0 & 78 & Dead  & -28848 & 1336 & 1224 & NA & WHITE                     & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 18\n",
       "\n",
       "| <!--/--> | analysis_id &lt;fct&gt; | barcode_final &lt;chr&gt; | project_id &lt;chr&gt; | barcode &lt;fct&gt; | disease &lt;fct&gt; | disease_name &lt;fct&gt; | tumor_tissue_site &lt;fct&gt; | histological_type &lt;fct&gt; | gender &lt;fct&gt; | days_to_initial_pathologic_diagnosis &lt;int&gt; | age_at_initial_pathologic_diagnosis &lt;int&gt; | vital_status &lt;fct&gt; | days_to_birth &lt;int&gt; | days_to_death &lt;int&gt; | days_to_last_followup &lt;int&gt; | patient_death_reason &lt;fct&gt; | race_list &lt;fct&gt; | treatment &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | fb47df4a-b9a7-46c6-9e8d-247c00057185 | TCGA-02-0047 | TCGA-GBM | TCGA-02-0047-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 78 | Dead  | -28759 |  448 |  448 | NA | WHITE                     | NA |\n",
       "| 2 | e1a99d0d-6b86-484d-8662-23c4378cde92 | TCGA-02-0055 | TCGA-GBM | TCGA-02-0055-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | FEMALE | 0 | 62 | Dead  | -22798 |   76 |   76 | NA | WHITE                     | NA |\n",
       "| 3 | 00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f | TCGA-02-2483 | TCGA-GBM | TCGA-02-2483-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 43 | Alive | -15964 |   NA |  466 | NA | ASIAN                     | NA |\n",
       "| 4 | f752e5d9-00a8-41f1-ae38-2b589f499088 | TCGA-02-2485 | TCGA-GBM | TCGA-02-2485-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 53 | Alive | -19494 |   NA |  470 | NA | BLACK OR AFRICAN AMERICAN | NA |\n",
       "| 5 | 50127941-3e32-46a5-956e-f55fbfc4080f | TCGA-02-2486 | TCGA-GBM | TCGA-02-2486-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 64 | Alive | -23394 |   NA |  493 | NA | WHITE                     | NA |\n",
       "| 6 | 2e13491b-c707-49e6-852c-ccd1d6252b00 | TCGA-04-1331 | TCGA-OV  | TCGA-04-1331-01A-01R-1569-13 | OV  | Ovarian serous cystadenocarcinoma | Ovary | Serous Cystadenocarcinoma       | FEMALE | 0 | 78 | Dead  | -28848 | 1336 | 1224 | NA | WHITE                     | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  analysis_id                          barcode_final project_id\n",
       "1 fb47df4a-b9a7-46c6-9e8d-247c00057185 TCGA-02-0047  TCGA-GBM  \n",
       "2 e1a99d0d-6b86-484d-8662-23c4378cde92 TCGA-02-0055  TCGA-GBM  \n",
       "3 00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f TCGA-02-2483  TCGA-GBM  \n",
       "4 f752e5d9-00a8-41f1-ae38-2b589f499088 TCGA-02-2485  TCGA-GBM  \n",
       "5 50127941-3e32-46a5-956e-f55fbfc4080f TCGA-02-2486  TCGA-GBM  \n",
       "6 2e13491b-c707-49e6-852c-ccd1d6252b00 TCGA-04-1331  TCGA-OV   \n",
       "  barcode                      disease disease_name                     \n",
       "1 TCGA-02-0047-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "2 TCGA-02-0055-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "3 TCGA-02-2483-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "4 TCGA-02-2485-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "5 TCGA-02-2486-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "6 TCGA-04-1331-01A-01R-1569-13 OV      Ovarian serous cystadenocarcinoma\n",
       "  tumor_tissue_site histological_type               gender\n",
       "1 Brain             Untreated primary (de novo) GBM MALE  \n",
       "2 Brain             Untreated primary (de novo) GBM FEMALE\n",
       "3 Brain             Untreated primary (de novo) GBM MALE  \n",
       "4 Brain             Untreated primary (de novo) GBM MALE  \n",
       "5 Brain             Untreated primary (de novo) GBM MALE  \n",
       "6 Ovary             Serous Cystadenocarcinoma       FEMALE\n",
       "  days_to_initial_pathologic_diagnosis age_at_initial_pathologic_diagnosis\n",
       "1 0                                    78                                 \n",
       "2 0                                    62                                 \n",
       "3 0                                    43                                 \n",
       "4 0                                    53                                 \n",
       "5 0                                    64                                 \n",
       "6 0                                    78                                 \n",
       "  vital_status days_to_birth days_to_death days_to_last_followup\n",
       "1 Dead         -28759         448           448                 \n",
       "2 Dead         -22798          76            76                 \n",
       "3 Alive        -15964          NA           466                 \n",
       "4 Alive        -19494          NA           470                 \n",
       "5 Alive        -23394          NA           493                 \n",
       "6 Dead         -28848        1336          1224                 \n",
       "  patient_death_reason race_list                 treatment\n",
       "1 NA                   WHITE                     NA       \n",
       "2 NA                   WHITE                     NA       \n",
       "3 NA                   ASIAN                     NA       \n",
       "4 NA                   BLACK OR AFRICAN AMERICAN NA       \n",
       "5 NA                   WHITE                     NA       \n",
       "6 NA                   WHITE                     NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# select main columns to be deposited in database\n",
    "data_id_with_clinical_db <- data_id_with_clinical[,c(\"analysis_id\",\"barcode_final\",\"project_id\",\n",
    "                                                    \"barcode\",\"disease\",\"disease_name\",\"tumor_tissue_site\",\n",
    "                                                    \"histological_type\",\"gender\",\n",
    "                                                    \"days_to_initial_pathologic_diagnosis\",\n",
    "                                                    \"age_at_initial_pathologic_diagnosis\",\n",
    "                                                    \"vital_status\",\"days_to_birth\",\"days_to_death\",\n",
    "                                                    \"days_to_last_followup\",\"patient_death_reason\",\n",
    "                                                     \"race_list\",\"treatment\")]\n",
    "head(data_id_with_clinical_db)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(data_id_with_clinical_db, \n",
    "            \"tcga_ccle_samples.barcode.with_targeted_analysis_ids.with_clinical_info.for_db.txt\",\n",
    "           sep=\"\\t\", row.names=FALSE, col.names=TRUE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### step5. generate unique genomic loci list for database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>10</td><td>101948358</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>10</td><td>101992683</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>10</td><td>101992848</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>10</td><td>102035116</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>10</td><td>102107838</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>10</td><td>102124544</td><td>G</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & V1 & V2 & V3\\\\\n",
       "  & <fct> & <int> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & 10 & 101948358 & C\\\\\n",
       "\t2 & 10 & 101992683 & T\\\\\n",
       "\t3 & 10 & 101992848 & T\\\\\n",
       "\t4 & 10 & 102035116 & C\\\\\n",
       "\t5 & 10 & 102107838 & C\\\\\n",
       "\t6 & 10 & 102124544 & G\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | V1 &lt;fct&gt; | V2 &lt;int&gt; | V3 &lt;fct&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | 10 | 101948358 | C |\n",
       "| 2 | 10 | 101992683 | T |\n",
       "| 3 | 10 | 101992848 | T |\n",
       "| 4 | 10 | 102035116 | C |\n",
       "| 5 | 10 | 102107838 | C |\n",
       "| 6 | 10 | 102124544 | G |\n",
       "\n"
      ],
      "text/plain": [
       "  V1 V2        V3\n",
       "1 10 101948358 C \n",
       "2 10 101992683 T \n",
       "3 10 101992848 T \n",
       "4 10 102035116 C \n",
       "5 10 102107838 C \n",
       "6 10 102124544 G "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "9825"
      ],
      "text/latex": [
       "9825"
      ],
      "text/markdown": [
       "9825"
      ],
      "text/plain": [
       "[1] 9825"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# before that,  file all_genomic_loci.txt has been generated, refer to 2_seggregate_RNA_mod_loci_for_each_sample.bash\n",
    "all_genomic_loci <- read.table(\"all_genomic_loci.txt\", sep = \"\\t\", head=FALSE)\n",
    "head(all_genomic_loci)\n",
    "nrow(all_genomic_loci)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "9825"
      ],
      "text/latex": [
       "9825"
      ],
      "text/markdown": [
       "9825"
      ],
      "text/plain": [
       "[1] 9825"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "all_genomic_loci_unique <- unique(all_genomic_loci)\n",
    "nrow(all_genomic_loci_unique)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>id</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>10</td><td>101948358</td><td>C</td><td>10-101948358-C</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>10</td><td>101992683</td><td>T</td><td>10-101992683-T</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>10</td><td>101992848</td><td>T</td><td>10-101992848-T</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>10</td><td>102035116</td><td>C</td><td>10-102035116-C</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>10</td><td>102107838</td><td>C</td><td>10-102107838-C</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>10</td><td>102124544</td><td>G</td><td>10-102124544-G</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & V1 & V2 & V3 & id\\\\\n",
       "  & <fct> & <int> & <fct> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 10 & 101948358 & C & 10-101948358-C\\\\\n",
       "\t2 & 10 & 101992683 & T & 10-101992683-T\\\\\n",
       "\t3 & 10 & 101992848 & T & 10-101992848-T\\\\\n",
       "\t4 & 10 & 102035116 & C & 10-102035116-C\\\\\n",
       "\t5 & 10 & 102107838 & C & 10-102107838-C\\\\\n",
       "\t6 & 10 & 102124544 & G & 10-102124544-G\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | V1 &lt;fct&gt; | V2 &lt;int&gt; | V3 &lt;fct&gt; | id &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 10 | 101948358 | C | 10-101948358-C |\n",
       "| 2 | 10 | 101992683 | T | 10-101992683-T |\n",
       "| 3 | 10 | 101992848 | T | 10-101992848-T |\n",
       "| 4 | 10 | 102035116 | C | 10-102035116-C |\n",
       "| 5 | 10 | 102107838 | C | 10-102107838-C |\n",
       "| 6 | 10 | 102124544 | G | 10-102124544-G |\n",
       "\n"
      ],
      "text/plain": [
       "  V1 V2        V3 id            \n",
       "1 10 101948358 C  10-101948358-C\n",
       "2 10 101992683 T  10-101992683-T\n",
       "3 10 101992848 T  10-101992848-T\n",
       "4 10 102035116 C  10-102035116-C\n",
       "5 10 102107838 C  10-102107838-C\n",
       "6 10 102124544 G  10-102124544-G"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "all_genomic_loci_unique$id <- apply(all_genomic_loci_unique, 1, function(x) paste(x, collapse = \"-\"))\n",
    "head(all_genomic_loci_unique)                     "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove spaces in ids\n",
    "all_genomic_loci_unique$id <- gsub(\" \", \"\", all_genomic_loci_unique$id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(all_genomic_loci_unique[,c(\"id\",\"V1\",\"V2\",\"V3\")], \"all_genomic_loci_uniq_with_ids.txt\",\n",
    "           sep=\"\\t\", col.names=FALSE, row.names=FALSE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Prepare the unique genomic loci in bed format for location annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>id</th><th scope=col>chr</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>10</td><td>101948358</td><td>C</td><td>10-101948358-C</td><td>chr10</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>10</td><td>101992683</td><td>T</td><td>10-101992683-T</td><td>chr10</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>10</td><td>101992848</td><td>T</td><td>10-101992848-T</td><td>chr10</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>10</td><td>102035116</td><td>C</td><td>10-102035116-C</td><td>chr10</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>10</td><td>102107838</td><td>C</td><td>10-102107838-C</td><td>chr10</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>10</td><td>102124544</td><td>G</td><td>10-102124544-G</td><td>chr10</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & V1 & V2 & V3 & id & chr\\\\\n",
       "  & <fct> & <int> & <fct> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 10 & 101948358 & C & 10-101948358-C & chr10\\\\\n",
       "\t2 & 10 & 101992683 & T & 10-101992683-T & chr10\\\\\n",
       "\t3 & 10 & 101992848 & T & 10-101992848-T & chr10\\\\\n",
       "\t4 & 10 & 102035116 & C & 10-102035116-C & chr10\\\\\n",
       "\t5 & 10 & 102107838 & C & 10-102107838-C & chr10\\\\\n",
       "\t6 & 10 & 102124544 & G & 10-102124544-G & chr10\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | V1 &lt;fct&gt; | V2 &lt;int&gt; | V3 &lt;fct&gt; | id &lt;chr&gt; | chr &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | 10 | 101948358 | C | 10-101948358-C | chr10 |\n",
       "| 2 | 10 | 101992683 | T | 10-101992683-T | chr10 |\n",
       "| 3 | 10 | 101992848 | T | 10-101992848-T | chr10 |\n",
       "| 4 | 10 | 102035116 | C | 10-102035116-C | chr10 |\n",
       "| 5 | 10 | 102107838 | C | 10-102107838-C | chr10 |\n",
       "| 6 | 10 | 102124544 | G | 10-102124544-G | chr10 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1 V2        V3 id             chr  \n",
       "1 10 101948358 C  10-101948358-C chr10\n",
       "2 10 101992683 T  10-101992683-T chr10\n",
       "3 10 101992848 T  10-101992848-T chr10\n",
       "4 10 102035116 C  10-102035116-C chr10\n",
       "5 10 102107838 C  10-102107838-C chr10\n",
       "6 10 102124544 G  10-102124544-G chr10"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "all_genomic_loci_unique$chr <- unlist(lapply(all_genomic_loci_unique$V1, function(x) paste0(\"chr\", x, sep=\"\")))\n",
    "head(all_genomic_loci_unique)         "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>V2</th><th scope=col>V2.1</th><th scope=col>id</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr10</td><td>101948358</td><td>101948358</td><td>10-101948358-C</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr10</td><td>101992683</td><td>101992683</td><td>10-101992683-T</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr10</td><td>101992848</td><td>101992848</td><td>10-101992848-T</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr10</td><td>102035116</td><td>102035116</td><td>10-102035116-C</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr10</td><td>102107838</td><td>102107838</td><td>10-102107838-C</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr10</td><td>102124544</td><td>102124544</td><td>10-102124544-G</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & chr & V2 & V2.1 & id\\\\\n",
       "  & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr10 & 101948358 & 101948358 & 10-101948358-C\\\\\n",
       "\t2 & chr10 & 101992683 & 101992683 & 10-101992683-T\\\\\n",
       "\t3 & chr10 & 101992848 & 101992848 & 10-101992848-T\\\\\n",
       "\t4 & chr10 & 102035116 & 102035116 & 10-102035116-C\\\\\n",
       "\t5 & chr10 & 102107838 & 102107838 & 10-102107838-C\\\\\n",
       "\t6 & chr10 & 102124544 & 102124544 & 10-102124544-G\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | V2 &lt;int&gt; | V2.1 &lt;int&gt; | id &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | chr10 | 101948358 | 101948358 | 10-101948358-C |\n",
       "| 2 | chr10 | 101992683 | 101992683 | 10-101992683-T |\n",
       "| 3 | chr10 | 101992848 | 101992848 | 10-101992848-T |\n",
       "| 4 | chr10 | 102035116 | 102035116 | 10-102035116-C |\n",
       "| 5 | chr10 | 102107838 | 102107838 | 10-102107838-C |\n",
       "| 6 | chr10 | 102124544 | 102124544 | 10-102124544-G |\n",
       "\n"
      ],
      "text/plain": [
       "  chr   V2        V2.1      id            \n",
       "1 chr10 101948358 101948358 10-101948358-C\n",
       "2 chr10 101992683 101992683 10-101992683-T\n",
       "3 chr10 101992848 101992848 10-101992848-T\n",
       "4 chr10 102035116 102035116 10-102035116-C\n",
       "5 chr10 102107838 102107838 10-102107838-C\n",
       "6 chr10 102124544 102124544 10-102124544-G"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "all_genomic_loci_unique_bed <- all_genomic_loci_unique[, c(\"chr\",\"V2\",\"V2\", \"id\")]\n",
    "head(all_genomic_loci_unique_bed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(all_genomic_loci_unique_bed, \"./annotate_all_genomic_loci/all_genomic_loci.bed\", \n",
    "           sep=\"\\t\", row.names=FALSE, col.names=FALSE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>PeakID..cmd.annotatePeaks.pl.all_genomic_loci.bed.hg19.</th><th scope=col>Distance.to.TSS</th><th scope=col>Gene.Name</th><th scope=col>Annotation</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>11-130014335-T</td><td>-15342</td><td>ST14    </td><td>non-coding </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>11-791543-T   </td><td> -1453</td><td>CEND1   </td><td>3' UTR     </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>7-72891715-C  </td><td> 43607</td><td>FZD9    </td><td>exon       </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>19-6414967-C  </td><td>  1555</td><td>MIR3940 </td><td>3' UTR     </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>11-57093539-T </td><td> -1113</td><td>TNKS1BP1</td><td>TTS        </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>5-98193906-C  </td><td> 68332</td><td>CHD1    </td><td>exon       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & PeakID..cmd.annotatePeaks.pl.all\\_genomic\\_loci.bed.hg19. & Distance.to.TSS & Gene.Name & Annotation\\\\\n",
       "  & <fct> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & 11-130014335-T & -15342 & ST14     & non-coding \\\\\n",
       "\t2 & 11-791543-T    &  -1453 & CEND1    & 3' UTR     \\\\\n",
       "\t3 & 7-72891715-C   &  43607 & FZD9     & exon       \\\\\n",
       "\t4 & 19-6414967-C   &   1555 & MIR3940  & 3' UTR     \\\\\n",
       "\t5 & 11-57093539-T  &  -1113 & TNKS1BP1 & TTS        \\\\\n",
       "\t6 & 5-98193906-C   &  68332 & CHD1     & exon       \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | PeakID..cmd.annotatePeaks.pl.all_genomic_loci.bed.hg19. &lt;fct&gt; | Distance.to.TSS &lt;int&gt; | Gene.Name &lt;fct&gt; | Annotation &lt;fct&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 11-130014335-T | -15342 | ST14     | non-coding  |\n",
       "| 2 | 11-791543-T    |  -1453 | CEND1    | 3' UTR      |\n",
       "| 3 | 7-72891715-C   |  43607 | FZD9     | exon        |\n",
       "| 4 | 19-6414967-C   |   1555 | MIR3940  | 3' UTR      |\n",
       "| 5 | 11-57093539-T  |  -1113 | TNKS1BP1 | TTS         |\n",
       "| 6 | 5-98193906-C   |  68332 | CHD1     | exon        |\n",
       "\n"
      ],
      "text/plain": [
       "  PeakID..cmd.annotatePeaks.pl.all_genomic_loci.bed.hg19. Distance.to.TSS\n",
       "1 11-130014335-T                                          -15342         \n",
       "2 11-791543-T                                              -1453         \n",
       "3 7-72891715-C                                             43607         \n",
       "4 19-6414967-C                                              1555         \n",
       "5 11-57093539-T                                            -1113         \n",
       "6 5-98193906-C                                             68332         \n",
       "  Gene.Name Annotation \n",
       "1 ST14      non-coding \n",
       "2 CEND1     3' UTR     \n",
       "3 FZD9      exon       \n",
       "4 MIR3940   3' UTR     \n",
       "5 TNKS1BP1  TTS        \n",
       "6 CHD1      exon       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# after location annotation using annotatePeaks.pl in HOMER package\n",
    "genomic_loci_annotate <- read.delim(\"./annotate_all_genomic_loci/annotation_of_all_genomic_loci_simplified.txt\",\n",
    "                                    sep=\"\\t\")\n",
    "head(genomic_loci_annotate)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(genomic_loci_annotate) <- c(\"id\", \"distance_to_tss\", \"gene_name\", \"annotation\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>id</th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>chr</th><th scope=col>distance_to_tss</th><th scope=col>gene_name</th><th scope=col>annotation</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1-100757026-G</td><td>1</td><td>100757026</td><td>G</td><td>chr1</td><td> 10229</td><td>MIR553</td><td>exon       </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1-101456149-C</td><td>1</td><td>101456149</td><td>C</td><td>chr1</td><td> 35191</td><td>DPH5  </td><td>exon       </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1-10502358-C </td><td>1</td><td> 10502358</td><td>C</td><td>chr1</td><td> -7730</td><td>CORT  </td><td>non-coding </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1-10522556-T </td><td>1</td><td> 10522556</td><td>T</td><td>chr1</td><td> 10030</td><td>DFFA  </td><td>3' UTR     </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1-10689932-A </td><td>1</td><td> 10689932</td><td>A</td><td>chr1</td><td>154925</td><td>PEX14 </td><td>exon       </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1-107599412-T</td><td>1</td><td>107599412</td><td>T</td><td>chr1</td><td>   116</td><td>PRMT6 </td><td>exon       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & id & V1 & V2 & V3 & chr & distance\\_to\\_tss & gene\\_name & annotation\\\\\n",
       "  & <chr> & <fct> & <int> & <fct> & <chr> & <int> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & 1-100757026-G & 1 & 100757026 & G & chr1 &  10229 & MIR553 & exon       \\\\\n",
       "\t2 & 1-101456149-C & 1 & 101456149 & C & chr1 &  35191 & DPH5   & exon       \\\\\n",
       "\t3 & 1-10502358-C  & 1 &  10502358 & C & chr1 &  -7730 & CORT   & non-coding \\\\\n",
       "\t4 & 1-10522556-T  & 1 &  10522556 & T & chr1 &  10030 & DFFA   & 3' UTR     \\\\\n",
       "\t5 & 1-10689932-A  & 1 &  10689932 & A & chr1 & 154925 & PEX14  & exon       \\\\\n",
       "\t6 & 1-107599412-T & 1 & 107599412 & T & chr1 &    116 & PRMT6  & exon       \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | id &lt;chr&gt; | V1 &lt;fct&gt; | V2 &lt;int&gt; | V3 &lt;fct&gt; | chr &lt;chr&gt; | distance_to_tss &lt;int&gt; | gene_name &lt;fct&gt; | annotation &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1-100757026-G | 1 | 100757026 | G | chr1 |  10229 | MIR553 | exon        |\n",
       "| 2 | 1-101456149-C | 1 | 101456149 | C | chr1 |  35191 | DPH5   | exon        |\n",
       "| 3 | 1-10502358-C  | 1 |  10502358 | C | chr1 |  -7730 | CORT   | non-coding  |\n",
       "| 4 | 1-10522556-T  | 1 |  10522556 | T | chr1 |  10030 | DFFA   | 3' UTR      |\n",
       "| 5 | 1-10689932-A  | 1 |  10689932 | A | chr1 | 154925 | PEX14  | exon        |\n",
       "| 6 | 1-107599412-T | 1 | 107599412 | T | chr1 |    116 | PRMT6  | exon        |\n",
       "\n"
      ],
      "text/plain": [
       "  id            V1 V2        V3 chr  distance_to_tss gene_name annotation \n",
       "1 1-100757026-G 1  100757026 G  chr1  10229          MIR553    exon       \n",
       "2 1-101456149-C 1  101456149 C  chr1  35191          DPH5      exon       \n",
       "3 1-10502358-C  1   10502358 C  chr1  -7730          CORT      non-coding \n",
       "4 1-10522556-T  1   10522556 T  chr1  10030          DFFA      3' UTR     \n",
       "5 1-10689932-A  1   10689932 A  chr1 154925          PEX14     exon       \n",
       "6 1-107599412-T 1  107599412 T  chr1    116          PRMT6     exon       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "all_genomic_loci_unique_with_anno <- merge(x = all_genomic_loci_unique, \n",
    "                                           y = genomic_loci_annotate,\n",
    "                                           by=\"id\")\n",
    "head(all_genomic_loci_unique_with_anno)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(all_genomic_loci_unique_with_anno[,c(\"id\",\"V1\",\"V2\",\"V3\",\"annotation\",\"gene_name\",\"distance_to_tss\")],\n",
    "           \"./all_genomic_loci_uniq_with_ids_and_annotations.txt\", sep=\"\\t\", row.names=FALSE, \n",
    "           col.names=FALSE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ModSite - generating data for dynamic interface of plots for each genomic loci accross all TCGA samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Registered S3 method overwritten by 'pryr':\n",
      "  method      from\n",
      "  print.bytes Rcpp\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(rbokeh)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>analysis_id</th><th scope=col>barcode_final</th><th scope=col>project_id</th><th scope=col>barcode</th><th scope=col>disease</th><th scope=col>disease_name</th><th scope=col>tumor_tissue_site</th><th scope=col>histological_type</th><th scope=col>gender</th><th scope=col>days_to_initial_pathologic_diagnosis</th><th scope=col>age_at_initial_pathologic_diagnosis</th><th scope=col>vital_status</th><th scope=col>days_to_birth</th><th scope=col>days_to_death</th><th scope=col>days_to_last_followup</th><th scope=col>patient_death_reason</th><th scope=col>race_list</th><th scope=col>treatment</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>fb47df4a-b9a7-46c6-9e8d-247c00057185</td><td>TCGA-02-0047</td><td>TCGA-GBM</td><td>TCGA-02-0047-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme</td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>78</td><td>Dead</td><td>-28759</td><td>448</td><td>448</td><td>NA</td><td>WHITE</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>e1a99d0d-6b86-484d-8662-23c4378cde92</td><td>TCGA-02-0055</td><td>TCGA-GBM</td><td>TCGA-02-0055-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme</td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>FEMALE</td><td>0</td><td>62</td><td>Dead</td><td>-22798</td><td> 76</td><td> 76</td><td>NA</td><td>WHITE</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 18\n",
       "\\begin{tabular}{r|llllllllllllllllll}\n",
       "  & analysis\\_id & barcode\\_final & project\\_id & barcode & disease & disease\\_name & tumor\\_tissue\\_site & histological\\_type & gender & days\\_to\\_initial\\_pathologic\\_diagnosis & age\\_at\\_initial\\_pathologic\\_diagnosis & vital\\_status & days\\_to\\_birth & days\\_to\\_death & days\\_to\\_last\\_followup & patient\\_death\\_reason & race\\_list & treatment\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & <fct> & <int> & <int> & <int> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & fb47df4a-b9a7-46c6-9e8d-247c00057185 & TCGA-02-0047 & TCGA-GBM & TCGA-02-0047-01A-01R-1849-01 & GBM & Glioblastoma multiforme & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 78 & Dead & -28759 & 448 & 448 & NA & WHITE & NA\\\\\n",
       "\t2 & e1a99d0d-6b86-484d-8662-23c4378cde92 & TCGA-02-0055 & TCGA-GBM & TCGA-02-0055-01A-01R-1849-01 & GBM & Glioblastoma multiforme & Brain & Untreated primary (de novo) GBM & FEMALE & 0 & 62 & Dead & -22798 &  76 &  76 & NA & WHITE & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 18\n",
       "\n",
       "| <!--/--> | analysis_id &lt;fct&gt; | barcode_final &lt;fct&gt; | project_id &lt;fct&gt; | barcode &lt;fct&gt; | disease &lt;fct&gt; | disease_name &lt;fct&gt; | tumor_tissue_site &lt;fct&gt; | histological_type &lt;fct&gt; | gender &lt;fct&gt; | days_to_initial_pathologic_diagnosis &lt;int&gt; | age_at_initial_pathologic_diagnosis &lt;int&gt; | vital_status &lt;fct&gt; | days_to_birth &lt;int&gt; | days_to_death &lt;int&gt; | days_to_last_followup &lt;int&gt; | patient_death_reason &lt;fct&gt; | race_list &lt;fct&gt; | treatment &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | fb47df4a-b9a7-46c6-9e8d-247c00057185 | TCGA-02-0047 | TCGA-GBM | TCGA-02-0047-01A-01R-1849-01 | GBM | Glioblastoma multiforme | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 78 | Dead | -28759 | 448 | 448 | NA | WHITE | NA |\n",
       "| 2 | e1a99d0d-6b86-484d-8662-23c4378cde92 | TCGA-02-0055 | TCGA-GBM | TCGA-02-0055-01A-01R-1849-01 | GBM | Glioblastoma multiforme | Brain | Untreated primary (de novo) GBM | FEMALE | 0 | 62 | Dead | -22798 |  76 |  76 | NA | WHITE | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  analysis_id                          barcode_final project_id\n",
       "1 fb47df4a-b9a7-46c6-9e8d-247c00057185 TCGA-02-0047  TCGA-GBM  \n",
       "2 e1a99d0d-6b86-484d-8662-23c4378cde92 TCGA-02-0055  TCGA-GBM  \n",
       "  barcode                      disease disease_name           \n",
       "1 TCGA-02-0047-01A-01R-1849-01 GBM     Glioblastoma multiforme\n",
       "2 TCGA-02-0055-01A-01R-1849-01 GBM     Glioblastoma multiforme\n",
       "  tumor_tissue_site histological_type               gender\n",
       "1 Brain             Untreated primary (de novo) GBM MALE  \n",
       "2 Brain             Untreated primary (de novo) GBM FEMALE\n",
       "  days_to_initial_pathologic_diagnosis age_at_initial_pathologic_diagnosis\n",
       "1 0                                    78                                 \n",
       "2 0                                    62                                 \n",
       "  vital_status days_to_birth days_to_death days_to_last_followup\n",
       "1 Dead         -28759        448           448                  \n",
       "2 Dead         -22798         76            76                  \n",
       "  patient_death_reason race_list treatment\n",
       "1 NA                   WHITE     NA       \n",
       "2 NA                   WHITE     NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# read clinical info \n",
    "clinical_info <- read.table(\"./tcga_ccle_samples.barcode.with_targeted_analysis_ids.with_clinical_info.for_db.txt\",\n",
    "                           header = TRUE, sep = \"\\t\")\n",
    "head(clinical_info, n=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>analysis_id</th><th scope=col>barcode_final</th><th scope=col>disease_name</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>fb47df4a-b9a7-46c6-9e8d-247c00057185</td><td>TCGA-02-0047</td><td>Glioblastoma multiforme</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>e1a99d0d-6b86-484d-8662-23c4378cde92</td><td>TCGA-02-0055</td><td>Glioblastoma multiforme</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & analysis\\_id & barcode\\_final & disease\\_name\\\\\n",
       "  & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & fb47df4a-b9a7-46c6-9e8d-247c00057185 & TCGA-02-0047 & Glioblastoma multiforme\\\\\n",
       "\t2 & e1a99d0d-6b86-484d-8662-23c4378cde92 & TCGA-02-0055 & Glioblastoma multiforme\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | analysis_id &lt;fct&gt; | barcode_final &lt;fct&gt; | disease_name &lt;fct&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | fb47df4a-b9a7-46c6-9e8d-247c00057185 | TCGA-02-0047 | Glioblastoma multiforme |\n",
       "| 2 | e1a99d0d-6b86-484d-8662-23c4378cde92 | TCGA-02-0055 | Glioblastoma multiforme |\n",
       "\n"
      ],
      "text/plain": [
       "  analysis_id                          barcode_final disease_name           \n",
       "1 fb47df4a-b9a7-46c6-9e8d-247c00057185 TCGA-02-0047  Glioblastoma multiforme\n",
       "2 e1a99d0d-6b86-484d-8662-23c4378cde92 TCGA-02-0055  Glioblastoma multiforme"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "clinical_info_sub <- clinical_info[,c(\"analysis_id\", \"barcode_final\", \"disease_name\")]\n",
    "head(clinical_info_sub, n = 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>id</th><th scope=col>chr</th><th scope=col>pos</th><th scope=col>ref</th><th scope=col>annotation</th><th scope=col>gene</th><th scope=col>distance</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1-100757026-G</td><td>1</td><td>100757026</td><td>G</td><td>exon      </td><td>MIR553</td><td> 10229</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1-101456149-C</td><td>1</td><td>101456149</td><td>C</td><td>exon      </td><td>DPH5  </td><td> 35191</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1-10502358-C </td><td>1</td><td> 10502358</td><td>C</td><td>non-coding</td><td>CORT  </td><td> -7730</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1-10522556-T </td><td>1</td><td> 10522556</td><td>T</td><td>3' UTR    </td><td>DFFA  </td><td> 10030</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1-10689932-A </td><td>1</td><td> 10689932</td><td>A</td><td>exon      </td><td>PEX14 </td><td>154925</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1-107599412-T</td><td>1</td><td>107599412</td><td>T</td><td>exon      </td><td>PRMT6 </td><td>   116</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & id & chr & pos & ref & annotation & gene & distance\\\\\n",
       "  & <fct> & <fct> & <int> & <fct> & <fct> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1-100757026-G & 1 & 100757026 & G & exon       & MIR553 &  10229\\\\\n",
       "\t2 & 1-101456149-C & 1 & 101456149 & C & exon       & DPH5   &  35191\\\\\n",
       "\t3 & 1-10502358-C  & 1 &  10502358 & C & non-coding & CORT   &  -7730\\\\\n",
       "\t4 & 1-10522556-T  & 1 &  10522556 & T & 3' UTR     & DFFA   &  10030\\\\\n",
       "\t5 & 1-10689932-A  & 1 &  10689932 & A & exon       & PEX14  & 154925\\\\\n",
       "\t6 & 1-107599412-T & 1 & 107599412 & T & exon       & PRMT6  &    116\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 7\n",
       "\n",
       "| <!--/--> | id &lt;fct&gt; | chr &lt;fct&gt; | pos &lt;int&gt; | ref &lt;fct&gt; | annotation &lt;fct&gt; | gene &lt;fct&gt; | distance &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1-100757026-G | 1 | 100757026 | G | exon       | MIR553 |  10229 |\n",
       "| 2 | 1-101456149-C | 1 | 101456149 | C | exon       | DPH5   |  35191 |\n",
       "| 3 | 1-10502358-C  | 1 |  10502358 | C | non-coding | CORT   |  -7730 |\n",
       "| 4 | 1-10522556-T  | 1 |  10522556 | T | 3' UTR     | DFFA   |  10030 |\n",
       "| 5 | 1-10689932-A  | 1 |  10689932 | A | exon       | PEX14  | 154925 |\n",
       "| 6 | 1-107599412-T | 1 | 107599412 | T | exon       | PRMT6  |    116 |\n",
       "\n"
      ],
      "text/plain": [
       "  id            chr pos       ref annotation gene   distance\n",
       "1 1-100757026-G 1   100757026 G   exon       MIR553  10229  \n",
       "2 1-101456149-C 1   101456149 C   exon       DPH5    35191  \n",
       "3 1-10502358-C  1    10502358 C   non-coding CORT    -7730  \n",
       "4 1-10522556-T  1    10522556 T   3' UTR     DFFA    10030  \n",
       "5 1-10689932-A  1    10689932 A   exon       PEX14  154925  \n",
       "6 1-107599412-T 1   107599412 T   exon       PRMT6     116  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# read all genomic loci\n",
    "all_genomic_loci <- read.delim(\"all_genomic_loci_uniq_with_ids_and_annotations.txt\", sep = \"\\t\", header = FALSE)\n",
    "colnames(all_genomic_loci) <- c(\"id\",\"chr\",\"pos\",\"ref\",\"annotation\",\"gene\",\"distance\")\n",
    "head(all_genomic_loci)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 100\n",
      "[1] 200\n",
      "[1] 300\n",
      "[1] 400\n",
      "[1] 500\n",
      "[1] 600\n",
      "[1] 700\n",
      "[1] 800\n",
      "[1] 900\n",
      "[1] 1000\n",
      "[1] 1100\n",
      "[1] 1200\n",
      "[1] 1300\n",
      "[1] 1400\n",
      "[1] 1500\n",
      "[1] 1600\n",
      "[1] 1700\n",
      "[1] 1800\n",
      "[1] 1900\n",
      "[1] 2000\n",
      "[1] 2100\n",
      "[1] 2200\n",
      "[1] 2300\n",
      "[1] 2400\n",
      "[1] 2500\n",
      "[1] 2600\n",
      "[1] 2700\n",
      "[1] 2800\n",
      "[1] 2900\n",
      "[1] 3000\n",
      "[1] 3100\n",
      "[1] 3200\n",
      "[1] 3300\n",
      "[1] 3400\n",
      "[1] 3500\n",
      "[1] 3600\n",
      "[1] 3700\n",
      "[1] 3800\n",
      "[1] 3900\n",
      "[1] 4000\n",
      "[1] 4100\n",
      "[1] 4200\n",
      "[1] 4300\n",
      "[1] 4400\n",
      "[1] 4500\n",
      "[1] 4600\n",
      "[1] 4700\n",
      "[1] 4800\n",
      "[1] 4900\n",
      "[1] 5000\n",
      "[1] 5100\n",
      "[1] 5200\n",
      "[1] 5300\n",
      "[1] 5400\n",
      "[1] 5500\n",
      "[1] 5600\n",
      "[1] 5700\n",
      "[1] 5800\n",
      "[1] 5900\n",
      "[1] 6000\n",
      "[1] 6100\n",
      "[1] 6200\n",
      "[1] 6300\n",
      "[1] 6400\n",
      "[1] 6500\n",
      "[1] 6600\n",
      "[1] 6700\n",
      "[1] 6800\n",
      "[1] 6900\n",
      "[1] 7000\n",
      "[1] 7100\n",
      "[1] 7200\n",
      "[1] 7300\n",
      "[1] 7400\n",
      "[1] 7500\n",
      "[1] 7600\n",
      "[1] 7700\n",
      "[1] 7800\n",
      "[1] 7900\n",
      "[1] 8000\n",
      "[1] 8100\n",
      "[1] 8200\n",
      "[1] 8300\n",
      "[1] 8400\n",
      "[1] 8500\n",
      "[1] 8600\n",
      "[1] 8700\n",
      "[1] 8800\n",
      "[1] 8900\n",
      "[1] 9000\n",
      "[1] 9100\n",
      "[1] 9200\n",
      "[1] 9300\n",
      "[1] 9400\n",
      "[1] 9500\n",
      "[1] 9600\n",
      "[1] 9700\n",
      "[1] 9800\n"
     ]
    }
   ],
   "source": [
    "# generate javascript file for the variant proportion at each genmoic locus\n",
    "for (i in 1:nrow(all_genomic_loci)){\n",
    "    filname_i <- paste0(\"./clean_data_selected_column/\",all_genomic_loci[i, \"chr\"],\n",
    "                        \"_\",all_genomic_loci[i,\"pos\"],\".completed\")\n",
    "    genomic_i <- read.table(filname_i, sep=\"\\t\", header = TRUE)\n",
    "    genomic_i_non0 <- genomic_i[which(genomic_i$variant_proportion > 0),]\n",
    "    genomic_i_non0_merge <- merge(x = genomic_i_non0, y = clinical_info_sub, \n",
    "                                by.x = \"analysis_ID\", by.y=\"analysis_id\")\n",
    "    \n",
    "    analysis_id <-  genomic_i_non0_merge[,\"analysis_ID\",drop=FALSE]\n",
    "    barcode <- genomic_i_non0_merge[,\"barcode_final\",drop=FALSE]\n",
    "    disease <- genomic_i_non0_merge[,\"disease_name\",drop=FALSE]\n",
    "    \n",
    "    x_max_round <- round(x = max(genomic_i_non0_merge$variant_proportion), digits = 1)\n",
    "    if (x_max_round >= max(genomic_i_non0_merge$variant_proportion)){\n",
    "        x_max <- x_max_round\n",
    "    } else {\n",
    "        x_max <- x_max_round + 0.05\n",
    "    }\n",
    "    p <- suppressWarnings(rbokeh::figure(xgrid=FALSE, ygrid = FALSE, xaxes = \"below\", yaxes = \"left\",\n",
    "              width = 500, height = 500, ylim=c(0, x_max),\n",
    "              logo = NULL, tools = c( \"pan\", \"wheel_zoom\",\"box_zoom\",\"reset\"), \n",
    "                                         xlab=\"\", ylab=\"variation proportion\") %>% \n",
    "    rbokeh::ly_points(x = seq(1,nrow(genomic_i_non0_merge),1), y=genomic_i_non0_merge$variant_proportion, \n",
    "                         alpha=0.5, size=5, hover = list(analysis_id, barcode, disease)))\n",
    "    \n",
    "    \n",
    "    suppressMessages(suppressWarnings(rbokeh::rbokeh2html(p, file = paste0(\"./variation_prop_plot_html/\",\n",
    "                                     all_genomic_loci[i, \"chr\"],\"_\",\n",
    "                                     all_genomic_loci[i,\"pos\"],\".html\"))))\n",
    "    \n",
    "    \n",
    "    index_html <- file(paste0(\"./variation_prop_plot_html/\",\n",
    "                                     all_genomic_loci[i, \"chr\"],\"_\",\n",
    "                                     all_genomic_loci[i,\"pos\"],\".html\"),\"r\")\n",
    "    modelid = \"\"\n",
    "    docid = \"\"\n",
    "    docs_json = \"\"\n",
    "    while (TRUE){\n",
    "        line = readLines(index_html, n=1, warn = FALSE)\n",
    "        if ( length(line) == 0 ) {\n",
    "            break\n",
    "        }\n",
    "        line = trimws(line)\n",
    "        if(startsWith(line, \"var modelid\")){\n",
    "            modelid = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "        if(startsWith(line, \"var docid\")){\n",
    "            docid = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "        if(startsWith(line, \"var docs_json\")){\n",
    "            docs_json = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "    }\n",
    "    close(index_html)\n",
    "    fileConn <- file(paste0(\"./variation_prop_plot_html_data/\",\n",
    "                                     all_genomic_loci[i, \"chr\"],\"_\",\n",
    "                                     all_genomic_loci[i,\"pos\"],\".js\"))\n",
    "    writeLines(c(\"Bokeh.$(function() {\",\n",
    "                 modelid, \n",
    "                 paste0(\"var elementid = 'modsite_modal_point_plot_\", \n",
    "                        all_genomic_loci[i, \"chr\"],\"_\",\n",
    "                        all_genomic_loci[i,\"pos\"],\"';\"),\n",
    "                 docid, \n",
    "                 docs_json,\n",
    "                \"var refkey = Object.keys(docs_json)[0]\",\n",
    "                \"var refs = docs_json[refkey].roots.references\",\n",
    "                \"function traverseObject(obj) {\",\n",
    "                \"  for (var key in obj) {\",\n",
    "                \"    if (obj[key].constructor === Object) {\",\n",
    "                \"      traverseObject(obj[key]);\",\n",
    "                \"    } else if (obj[key].constructor === Array) {\",\n",
    "                \"      for (var i = 0; i < obj[key].length; i++) {\",\n",
    "                \"        if (obj[key][i] === null)\",\n",
    "                \"        obj[key][i] = NaN;\",\n",
    "                \"      };\",\n",
    "                \"    }\",\n",
    "                \"  };\",\n",
    "                \"}\",\n",
    "                \"for (var i = 0; i < refs.length; i++) {\",\n",
    "                \"  if (refs[i].type === 'ColumnDataSource')\",\n",
    "                \"    traverseObject(refs[i].attributes.data);\",\n",
    "                \"};\",\n",
    "                \"var render_items = [{\",\n",
    "                \"  'docid': docid,\",\n",
    "                \"  'elementid': elementid,\",\n",
    "                \"  'modelid': modelid\",\n",
    "                \"}];\",\n",
    "                \"Bokeh.set_log_level('info');\",\n",
    "                \"Bokeh.embed.embed_items(docs_json, render_items);\",\n",
    "                \"});\"), fileConn)\n",
    "    close(fileConn)\n",
    "    \n",
    "    if (i%%100==0){print(i)}\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 100\n",
      "[1] 200\n",
      "[1] 300\n",
      "[1] 400\n",
      "[1] 500\n",
      "[1] 600\n",
      "[1] 700\n",
      "[1] 800\n",
      "[1] 900\n",
      "[1] 1000\n",
      "[1] 1100\n",
      "[1] 1200\n",
      "[1] 1300\n",
      "[1] 1400\n",
      "[1] 1500\n",
      "[1] 1600\n",
      "[1] 1700\n",
      "[1] 1800\n",
      "[1] 1900\n",
      "[1] 2000\n",
      "[1] 2100\n",
      "[1] 2200\n",
      "[1] 2300\n",
      "[1] 2400\n",
      "[1] 2500\n",
      "[1] 2600\n",
      "[1] 2700\n",
      "[1] 2800\n",
      "[1] 2900\n",
      "[1] 3000\n",
      "[1] 3100\n",
      "[1] 3200\n",
      "[1] 3300\n",
      "[1] 3400\n",
      "[1] 3500\n",
      "[1] 3600\n",
      "[1] 3700\n",
      "[1] 3800\n",
      "[1] 3900\n",
      "[1] 4000\n",
      "[1] 4100\n",
      "[1] 4200\n",
      "[1] 4300\n",
      "[1] 4400\n",
      "[1] 4500\n",
      "[1] 4600\n",
      "[1] 4700\n",
      "[1] 4800\n",
      "[1] 4900\n",
      "[1] 5000\n",
      "[1] 5100\n",
      "[1] 5200\n",
      "[1] 5300\n",
      "[1] 5400\n",
      "[1] 5500\n",
      "[1] 5600\n",
      "[1] 5700\n",
      "[1] 5800\n",
      "[1] 5900\n",
      "[1] 6000\n",
      "[1] 6100\n",
      "[1] 6200\n",
      "[1] 6300\n",
      "[1] 6400\n",
      "[1] 6500\n",
      "[1] 6600\n",
      "[1] 6700\n",
      "[1] 6800\n",
      "[1] 6900\n",
      "[1] 7000\n",
      "[1] 7100\n",
      "[1] 7200\n",
      "[1] 7300\n",
      "[1] 7400\n",
      "[1] 7500\n",
      "[1] 7600\n",
      "[1] 7700\n",
      "[1] 7800\n",
      "[1] 7900\n",
      "[1] 8000\n",
      "[1] 8100\n",
      "[1] 8200\n",
      "[1] 8300\n",
      "[1] 8400\n",
      "[1] 8500\n",
      "[1] 8600\n",
      "[1] 8700\n",
      "[1] 8800\n",
      "[1] 8900\n",
      "[1] 9000\n",
      "[1] 9100\n",
      "[1] 9200\n",
      "[1] 9300\n",
      "[1] 9400\n",
      "[1] 9500\n",
      "[1] 9600\n",
      "[1] 9700\n",
      "[1] 9800\n"
     ]
    }
   ],
   "source": [
    "# generate javascript file for the deletion proportion at each genmoic locus\n",
    "for (i in 1:nrow(all_genomic_loci)){\n",
    "    filname_i <- paste0(\"./clean_data_selected_column/\",all_genomic_loci[i, \"chr\"],\n",
    "                        \"_\",all_genomic_loci[i,\"pos\"],\".completed\")\n",
    "    genomic_i <- read.table(filname_i, sep=\"\\t\", header = TRUE)\n",
    "    genomic_i_non0 <- genomic_i[which(genomic_i$variant_proportion > 0),]\n",
    "    genomic_i_non0_merge <- merge(x = genomic_i_non0, y = clinical_info_sub, \n",
    "                                by.x = \"analysis_ID\", by.y=\"analysis_id\")\n",
    "    \n",
    "    analysis_id <-  genomic_i_non0_merge[,\"analysis_ID\",drop=FALSE]\n",
    "    barcode <- genomic_i_non0_merge[,\"barcode_final\",drop=FALSE]\n",
    "    disease <- genomic_i_non0_merge[,\"disease_name\",drop=FALSE]\n",
    "    \n",
    "    x_max_round <- round(x = max(genomic_i_non0_merge$deletion_proportion), digits = 1)\n",
    "    if (x_max_round >= max(genomic_i_non0_merge$deletion_proportion)){\n",
    "        x_max <- x_max_round\n",
    "    } else {\n",
    "        x_max <- x_max_round + 0.05\n",
    "    }\n",
    "    p <- suppressWarnings(rbokeh::figure(xgrid=FALSE, ygrid = FALSE, xaxes = \"below\", yaxes = \"left\",\n",
    "              width = 500, height = 500, ylim=c(0, x_max),\n",
    "              logo = NULL, tools = c( \"pan\", \"wheel_zoom\",\"box_zoom\",\"reset\"), \n",
    "                                         xlab=\"\", ylab=\"deletion proportion\") %>% \n",
    "    rbokeh::ly_points(x = seq(1,nrow(genomic_i_non0_merge),1), y=genomic_i_non0_merge$deletion_proportion, \n",
    "                         alpha=0.5, size=5, hover = list(analysis_id, barcode, disease)))\n",
    "    \n",
    "    \n",
    "    suppressMessages(suppressWarnings(rbokeh::rbokeh2html(p, file = paste0(\"./deletion_prop_plot_html/\",\n",
    "                                     all_genomic_loci[i, \"chr\"],\"_\",\n",
    "                                     all_genomic_loci[i,\"pos\"],\".html\"))))\n",
    "    \n",
    "    \n",
    "    index_html <- file(paste0(\"./deletion_prop_plot_html/\",\n",
    "                                     all_genomic_loci[i, \"chr\"],\"_\",\n",
    "                                     all_genomic_loci[i,\"pos\"],\".html\"),\"r\")\n",
    "    modelid = \"\"\n",
    "    docid = \"\"\n",
    "    docs_json = \"\"\n",
    "    while (TRUE){\n",
    "        line = readLines(index_html, n=1, warn = FALSE)\n",
    "        if ( length(line) == 0 ) {\n",
    "            break\n",
    "        }\n",
    "        line = trimws(line)\n",
    "        if(startsWith(line, \"var modelid\")){\n",
    "            modelid = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "        if(startsWith(line, \"var docid\")){\n",
    "            docid = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "        if(startsWith(line, \"var docs_json\")){\n",
    "            docs_json = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "    }\n",
    "    close(index_html)\n",
    "    fileConn <- file(paste0(\"./deletion_prop_plot_html_data/\",\n",
    "                                     all_genomic_loci[i, \"chr\"],\"_\",\n",
    "                                     all_genomic_loci[i,\"pos\"],\".js\"))\n",
    "    writeLines(c(\"Bokeh.$(function() {\",\n",
    "                 modelid, \n",
    "                 paste0(\"var elementid = 'modsite_modal_point_plot_deletion_\", \n",
    "                        all_genomic_loci[i, \"chr\"],\"_\",\n",
    "                        all_genomic_loci[i,\"pos\"],\"';\"),\n",
    "                 docid, \n",
    "                 docs_json,\n",
    "                \"var refkey = Object.keys(docs_json)[0]\",\n",
    "                \"var refs = docs_json[refkey].roots.references\",\n",
    "                \"function traverseObject(obj) {\",\n",
    "                \"  for (var key in obj) {\",\n",
    "                \"    if (obj[key].constructor === Object) {\",\n",
    "                \"      traverseObject(obj[key]);\",\n",
    "                \"    } else if (obj[key].constructor === Array) {\",\n",
    "                \"      for (var i = 0; i < obj[key].length; i++) {\",\n",
    "                \"        if (obj[key][i] === null)\",\n",
    "                \"        obj[key][i] = NaN;\",\n",
    "                \"      };\",\n",
    "                \"    }\",\n",
    "                \"  };\",\n",
    "                \"}\",\n",
    "                \"for (var i = 0; i < refs.length; i++) {\",\n",
    "                \"  if (refs[i].type === 'ColumnDataSource')\",\n",
    "                \"    traverseObject(refs[i].attributes.data);\",\n",
    "                \"};\",\n",
    "                \"var render_items = [{\",\n",
    "                \"  'docid': docid,\",\n",
    "                \"  'elementid': elementid,\",\n",
    "                \"  'modelid': modelid\",\n",
    "                \"}];\",\n",
    "                \"Bokeh.set_log_level('info');\",\n",
    "                \"Bokeh.embed.embed_items(docs_json, render_items);\",\n",
    "                \"});\"), fileConn)\n",
    "    close(fileConn)\n",
    "    \n",
    "    if (i%%100==0){print(i)}\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## generate survival plot \n",
    "###  genomic locus with variation proportion more than 0.01 is defined as potential RNA modification site\n",
    "### survival plot with and without RNA modifications"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(survival)\n",
    "library(survminer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read clinical info \n",
    "clinical_info <- read.table(\"./tcga_ccle_samples.barcode.with_targeted_analysis_ids.with_clinical_info.for_db.txt\",\n",
    "                           header = TRUE, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_info_sub <- clinical_info[,c(\"analysis_id\", \"barcode_final\", \n",
    "                                      \"disease\", \"disease_name\",\n",
    "                                     \"vital_status\", \"days_to_last_followup\",\n",
    "                                      \"days_to_death\", \"barcode\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get tumor only\n",
    "clinical_info_sub$normal_dis_id <- unlist(lapply(as.character(clinical_info_sub$barcode), \n",
    "                                                 function(x)  unlist(strsplit(x, split = \"-\"))[4] ))\n",
    "\n",
    "clinical_info_sub$first_code <- unlist(lapply(clinical_info_sub$normal_dis_id, \n",
    "                                             function(x) unlist(strsplit(x, split = \"\"))[1]  ))\n",
    "                                              \n",
    "clinical_info_sub_tumor <- clinical_info_sub[which(clinical_info_sub$first_code == \"0\"),]\n",
    "\n",
    "clinical_info_sub <- clinical_info_sub_tumor[,c(\"analysis_id\", \"barcode_final\", \n",
    "                                      \"disease\", \"disease_name\",\n",
    "                                     \"vital_status\", \"days_to_last_followup\",\n",
    "                                               \"days_to_death\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read all genomic loci\n",
    "all_genomic_loci <- read.delim(\"all_genomic_loci_uniq_with_ids_and_annotations.txt\", sep = \"\\t\", header = FALSE)\n",
    "colnames(all_genomic_loci) <- c(\"id\",\"chr\",\"pos\",\"ref\",\"annotation\",\"gene\",\"distance\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# generate survival plots without filtration\n",
    "for (i in 1:nrow(all_genomic_loci)){\n",
    "    dir.create(paste0(\"./survival_results_for_each_genomic_locus/chr\",all_genomic_loci[i, \"chr\"],\n",
    "                     \"_\", all_genomic_loci[i,\"pos\"]))\n",
    "    filname_i <- paste0(\"./clean_data_selected_column/\",all_genomic_loci[i, \"chr\"],\n",
    "                        \"_\",all_genomic_loci[i,\"pos\"],\".completed\")\n",
    "    genomic_i <- read.table(filname_i, sep=\"\\t\", header = TRUE)\n",
    "    genomic_i_non0 <- genomic_i[which(genomic_i$variant_proportion >= 0.01),]\n",
    "    genomic_i_non0_merge <- merge(x = genomic_i_non0, y = clinical_info_sub, \n",
    "                                by.x = \"analysis_ID\", by.y=\"analysis_id\")\n",
    "    \n",
    "    dis_uniq <- unique(clinical_info_sub$disease)\n",
    "    all_dis <- data.frame( dis_uniq[order(dis_uniq)] )\n",
    "    colnames(all_dis) <- \"all_dis\"\n",
    "    all_dis$number <- 0\n",
    "    for (dis in unique(genomic_i_non0_merge$disease)){\n",
    "        \n",
    "        genomic_i_non0_merge_dis <- genomic_i_non0_merge[which(genomic_i_non0_merge$disease == dis &\n",
    "                                                       genomic_i_non0_merge$vital_status %in% c(\"Alive\",\"Dead\")),]\n",
    "        \n",
    "        clinical_info_dis <- clinical_info_sub[which(clinical_info_sub$disease == dis &\n",
    "                                         clinical_info_sub$vital_status %in% c(\"Alive\",\"Dead\") ),]\n",
    "        clinical_info_dis[which(clinical_info_dis$vital_status == \"Dead\"), \n",
    "                  \"days_to_last_followup\"] <- clinical_info_dis[which(clinical_info_dis$vital_status == \"Dead\"), \n",
    "                                                                \"days_to_death\"]\n",
    "        clinical_info_dis_noNA <- clinical_info_dis[!is.na(clinical_info_dis$days_to_last_followup), ]\n",
    "        \n",
    "        # all_dis\n",
    "        all_dis[which(all_dis$all_dis == dis), \n",
    "                \"number\"] <- nrow(clinical_info_dis_noNA[which(clinical_info_dis_noNA$analysis_id %in%\n",
    "                                                               genomic_i_non0_merge_dis$analysis_ID),])\n",
    "\n",
    "        clinical_info_dis_noNA$event <- 0\n",
    "        clinical_info_dis_noNA[which(clinical_info_dis_noNA$vital_status == \"Dead\"),\"event\"] <- 1\n",
    "        clinical_info_dis_noNA$type <- \"without_mod\"\n",
    "        clinical_info_dis_noNA[which(clinical_info_dis_noNA$analysis_id %in% genomic_i_non0_merge_dis$analysis_ID),\"type\"] <- \"with_mod\"\n",
    "        num_without_mod <- nrow(clinical_info_dis_noNA[which(clinical_info_dis_noNA$type == \"without_mod\"),])\n",
    "        num_with_mod <- nrow(clinical_info_dis_noNA[which(clinical_info_dis_noNA$type == \"with_mod\"),])\n",
    "        \n",
    "        clinical_info_dis_noNA$type <- factor(clinical_info_dis_noNA$type)\n",
    "\n",
    "        \n",
    "        if (num_without_mod>=30 &  num_with_mod>=30 ){\n",
    "            \n",
    "            \n",
    "            png_name <- paste0(\"./survival_results_for_each_genomic_locus/chr\",all_genomic_loci[i, \"chr\"],\n",
    "                     \"_\", all_genomic_loci[i,\"pos\"], \"/survival_analysis_for_\", dis,\".png\")\n",
    "            \n",
    "            surv_object <- Surv(time = clinical_info_dis_noNA$days_to_last_followup, \n",
    "                    event = clinical_info_dis_noNA$event)\n",
    "            fit1 <- survfit(surv_object ~ type, data = clinical_info_dis_noNA)\n",
    "            title_name <- paste0(\"with_mod num = \", num_with_mod,\n",
    "                                \", without_mod num = \", num_without_mod)\n",
    "            \n",
    "            #png(png_name, units=\"px\", width=1200, height=1200, res=150)\n",
    "            survp <- ggsurvplot(fit1, data = clinical_info_dis_noNA, pval = TRUE, title=title_name)\n",
    "            ggsave(file = png_name, print(survp))\n",
    "            #dev.off()\n",
    "            \n",
    "        }        \n",
    "    }\n",
    "    \n",
    "    dis_filename <- paste0(\"./survival_results_for_each_genomic_locus/chr\",all_genomic_loci[i, \"chr\"],\n",
    "                     \"_\", all_genomic_loci[i,\"pos\"], \"/sample_number_with_mod_in_diseases.txt\")\n",
    "    write.table(all_dis, dis_filename, sep = \"\\t\", row.names = FALSE, col.names = FALSE, quote = FALSE)\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ModPatient - generating data for dynamic interface of plots for each TCGA samples accross all genomic loci"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Registered S3 method overwritten by 'pryr':\n",
      "  method      from\n",
      "  print.bytes Rcpp\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(rbokeh)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>analysis_id</th><th scope=col>barcode_final</th><th scope=col>project_id</th><th scope=col>barcode</th><th scope=col>disease</th><th scope=col>disease_name</th><th scope=col>tumor_tissue_site</th><th scope=col>histological_type</th><th scope=col>gender</th><th scope=col>days_to_initial_pathologic_diagnosis</th><th scope=col>age_at_initial_pathologic_diagnosis</th><th scope=col>vital_status</th><th scope=col>days_to_birth</th><th scope=col>days_to_death</th><th scope=col>days_to_last_followup</th><th scope=col>patient_death_reason</th><th scope=col>race_list</th><th scope=col>treatment</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>fb47df4a-b9a7-46c6-9e8d-247c00057185</td><td>TCGA-02-0047</td><td>TCGA-GBM</td><td>TCGA-02-0047-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>78</td><td>Dead </td><td>-28759</td><td> 448</td><td> 448</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>e1a99d0d-6b86-484d-8662-23c4378cde92</td><td>TCGA-02-0055</td><td>TCGA-GBM</td><td>TCGA-02-0055-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>FEMALE</td><td>0</td><td>62</td><td>Dead </td><td>-22798</td><td>  76</td><td>  76</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f</td><td>TCGA-02-2483</td><td>TCGA-GBM</td><td>TCGA-02-2483-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>43</td><td>Alive</td><td>-15964</td><td>  NA</td><td> 466</td><td>NA</td><td>ASIAN                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>f752e5d9-00a8-41f1-ae38-2b589f499088</td><td>TCGA-02-2485</td><td>TCGA-GBM</td><td>TCGA-02-2485-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>53</td><td>Alive</td><td>-19494</td><td>  NA</td><td> 470</td><td>NA</td><td>BLACK OR AFRICAN AMERICAN</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>50127941-3e32-46a5-956e-f55fbfc4080f</td><td>TCGA-02-2486</td><td>TCGA-GBM</td><td>TCGA-02-2486-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>64</td><td>Alive</td><td>-23394</td><td>  NA</td><td> 493</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>2e13491b-c707-49e6-852c-ccd1d6252b00</td><td>TCGA-04-1331</td><td>TCGA-OV </td><td>TCGA-04-1331-01A-01R-1569-13</td><td>OV </td><td>Ovarian serous cystadenocarcinoma</td><td>Ovary</td><td>Serous Cystadenocarcinoma      </td><td>FEMALE</td><td>0</td><td>78</td><td>Dead </td><td>-28848</td><td>1336</td><td>1224</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 18\n",
       "\\begin{tabular}{r|llllllllllllllllll}\n",
       "  & analysis\\_id & barcode\\_final & project\\_id & barcode & disease & disease\\_name & tumor\\_tissue\\_site & histological\\_type & gender & days\\_to\\_initial\\_pathologic\\_diagnosis & age\\_at\\_initial\\_pathologic\\_diagnosis & vital\\_status & days\\_to\\_birth & days\\_to\\_death & days\\_to\\_last\\_followup & patient\\_death\\_reason & race\\_list & treatment\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & <fct> & <int> & <int> & <int> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & fb47df4a-b9a7-46c6-9e8d-247c00057185 & TCGA-02-0047 & TCGA-GBM & TCGA-02-0047-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 78 & Dead  & -28759 &  448 &  448 & NA & WHITE                     & NA\\\\\n",
       "\t2 & e1a99d0d-6b86-484d-8662-23c4378cde92 & TCGA-02-0055 & TCGA-GBM & TCGA-02-0055-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & FEMALE & 0 & 62 & Dead  & -22798 &   76 &   76 & NA & WHITE                     & NA\\\\\n",
       "\t3 & 00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f & TCGA-02-2483 & TCGA-GBM & TCGA-02-2483-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 43 & Alive & -15964 &   NA &  466 & NA & ASIAN                     & NA\\\\\n",
       "\t4 & f752e5d9-00a8-41f1-ae38-2b589f499088 & TCGA-02-2485 & TCGA-GBM & TCGA-02-2485-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 53 & Alive & -19494 &   NA &  470 & NA & BLACK OR AFRICAN AMERICAN & NA\\\\\n",
       "\t5 & 50127941-3e32-46a5-956e-f55fbfc4080f & TCGA-02-2486 & TCGA-GBM & TCGA-02-2486-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 64 & Alive & -23394 &   NA &  493 & NA & WHITE                     & NA\\\\\n",
       "\t6 & 2e13491b-c707-49e6-852c-ccd1d6252b00 & TCGA-04-1331 & TCGA-OV  & TCGA-04-1331-01A-01R-1569-13 & OV  & Ovarian serous cystadenocarcinoma & Ovary & Serous Cystadenocarcinoma       & FEMALE & 0 & 78 & Dead  & -28848 & 1336 & 1224 & NA & WHITE                     & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 18\n",
       "\n",
       "| <!--/--> | analysis_id &lt;fct&gt; | barcode_final &lt;fct&gt; | project_id &lt;fct&gt; | barcode &lt;fct&gt; | disease &lt;fct&gt; | disease_name &lt;fct&gt; | tumor_tissue_site &lt;fct&gt; | histological_type &lt;fct&gt; | gender &lt;fct&gt; | days_to_initial_pathologic_diagnosis &lt;int&gt; | age_at_initial_pathologic_diagnosis &lt;int&gt; | vital_status &lt;fct&gt; | days_to_birth &lt;int&gt; | days_to_death &lt;int&gt; | days_to_last_followup &lt;int&gt; | patient_death_reason &lt;fct&gt; | race_list &lt;fct&gt; | treatment &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | fb47df4a-b9a7-46c6-9e8d-247c00057185 | TCGA-02-0047 | TCGA-GBM | TCGA-02-0047-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 78 | Dead  | -28759 |  448 |  448 | NA | WHITE                     | NA |\n",
       "| 2 | e1a99d0d-6b86-484d-8662-23c4378cde92 | TCGA-02-0055 | TCGA-GBM | TCGA-02-0055-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | FEMALE | 0 | 62 | Dead  | -22798 |   76 |   76 | NA | WHITE                     | NA |\n",
       "| 3 | 00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f | TCGA-02-2483 | TCGA-GBM | TCGA-02-2483-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 43 | Alive | -15964 |   NA |  466 | NA | ASIAN                     | NA |\n",
       "| 4 | f752e5d9-00a8-41f1-ae38-2b589f499088 | TCGA-02-2485 | TCGA-GBM | TCGA-02-2485-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 53 | Alive | -19494 |   NA |  470 | NA | BLACK OR AFRICAN AMERICAN | NA |\n",
       "| 5 | 50127941-3e32-46a5-956e-f55fbfc4080f | TCGA-02-2486 | TCGA-GBM | TCGA-02-2486-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 64 | Alive | -23394 |   NA |  493 | NA | WHITE                     | NA |\n",
       "| 6 | 2e13491b-c707-49e6-852c-ccd1d6252b00 | TCGA-04-1331 | TCGA-OV  | TCGA-04-1331-01A-01R-1569-13 | OV  | Ovarian serous cystadenocarcinoma | Ovary | Serous Cystadenocarcinoma       | FEMALE | 0 | 78 | Dead  | -28848 | 1336 | 1224 | NA | WHITE                     | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  analysis_id                          barcode_final project_id\n",
       "1 fb47df4a-b9a7-46c6-9e8d-247c00057185 TCGA-02-0047  TCGA-GBM  \n",
       "2 e1a99d0d-6b86-484d-8662-23c4378cde92 TCGA-02-0055  TCGA-GBM  \n",
       "3 00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f TCGA-02-2483  TCGA-GBM  \n",
       "4 f752e5d9-00a8-41f1-ae38-2b589f499088 TCGA-02-2485  TCGA-GBM  \n",
       "5 50127941-3e32-46a5-956e-f55fbfc4080f TCGA-02-2486  TCGA-GBM  \n",
       "6 2e13491b-c707-49e6-852c-ccd1d6252b00 TCGA-04-1331  TCGA-OV   \n",
       "  barcode                      disease disease_name                     \n",
       "1 TCGA-02-0047-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "2 TCGA-02-0055-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "3 TCGA-02-2483-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "4 TCGA-02-2485-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "5 TCGA-02-2486-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "6 TCGA-04-1331-01A-01R-1569-13 OV      Ovarian serous cystadenocarcinoma\n",
       "  tumor_tissue_site histological_type               gender\n",
       "1 Brain             Untreated primary (de novo) GBM MALE  \n",
       "2 Brain             Untreated primary (de novo) GBM FEMALE\n",
       "3 Brain             Untreated primary (de novo) GBM MALE  \n",
       "4 Brain             Untreated primary (de novo) GBM MALE  \n",
       "5 Brain             Untreated primary (de novo) GBM MALE  \n",
       "6 Ovary             Serous Cystadenocarcinoma       FEMALE\n",
       "  days_to_initial_pathologic_diagnosis age_at_initial_pathologic_diagnosis\n",
       "1 0                                    78                                 \n",
       "2 0                                    62                                 \n",
       "3 0                                    43                                 \n",
       "4 0                                    53                                 \n",
       "5 0                                    64                                 \n",
       "6 0                                    78                                 \n",
       "  vital_status days_to_birth days_to_death days_to_last_followup\n",
       "1 Dead         -28759         448           448                 \n",
       "2 Dead         -22798          76            76                 \n",
       "3 Alive        -15964          NA           466                 \n",
       "4 Alive        -19494          NA           470                 \n",
       "5 Alive        -23394          NA           493                 \n",
       "6 Dead         -28848        1336          1224                 \n",
       "  patient_death_reason race_list                 treatment\n",
       "1 NA                   WHITE                     NA       \n",
       "2 NA                   WHITE                     NA       \n",
       "3 NA                   ASIAN                     NA       \n",
       "4 NA                   BLACK OR AFRICAN AMERICAN NA       \n",
       "5 NA                   WHITE                     NA       \n",
       "6 NA                   WHITE                     NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "all_tcga <- read.table(\"tcga_ccle_samples.barcode.with_targeted_analysis_ids.with_clinical_info.for_db.txt\",\n",
    "                      sep=\"\\t\", header=TRUE)\n",
    "head(all_tcga)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 100\n",
      "[1] 200\n",
      "[1] 300\n",
      "[1] 400\n",
      "[1] 500\n",
      "[1] 600\n",
      "[1] 700\n",
      "[1] 800\n",
      "[1] 900\n",
      "[1] 1000\n",
      "[1] 1100\n",
      "[1] 1200\n",
      "[1] 1300\n",
      "[1] 1400\n",
      "[1] 1500\n",
      "[1] 1600\n",
      "[1] 1700\n",
      "[1] 1800\n",
      "[1] 1900\n",
      "[1] 2000\n",
      "[1] 2100\n",
      "[1] 2200\n",
      "[1] 2300\n",
      "[1] 2400\n",
      "[1] 2500\n",
      "[1] 2600\n",
      "[1] 2700\n",
      "[1] 2800\n",
      "[1] 2900\n",
      "[1] 3000\n",
      "[1] 3100\n",
      "[1] 3200\n",
      "[1] 3300\n",
      "[1] 3400\n",
      "[1] 3500\n",
      "[1] 3600\n",
      "[1] 3700\n",
      "[1] 3800\n",
      "[1] 3900\n",
      "[1] 4000\n",
      "[1] 4100\n",
      "[1] 4200\n",
      "[1] 4300\n",
      "[1] 4400\n",
      "[1] 4500\n",
      "[1] 4600\n",
      "[1] 4700\n",
      "[1] 4800\n",
      "[1] 4900\n",
      "[1] 5000\n",
      "[1] 5100\n",
      "[1] 5200\n",
      "[1] 5300\n",
      "[1] 5400\n",
      "[1] 5500\n",
      "[1] 5600\n",
      "[1] 5700\n",
      "[1] 5800\n",
      "[1] 5900\n",
      "[1] 6000\n",
      "[1] 6100\n",
      "[1] 6200\n",
      "[1] 6300\n",
      "[1] 6400\n",
      "[1] 6500\n",
      "[1] 6600\n",
      "[1] 6700\n",
      "[1] 6800\n",
      "[1] 6900\n",
      "[1] 7000\n",
      "[1] 7100\n",
      "[1] 7200\n",
      "[1] 7300\n",
      "[1] 7400\n",
      "[1] 7500\n",
      "[1] 7600\n",
      "[1] 7700\n",
      "[1] 7800\n",
      "[1] 7900\n",
      "[1] 8000\n",
      "[1] 8100\n",
      "[1] 8200\n",
      "[1] 8300\n",
      "[1] 8400\n",
      "[1] 8500\n",
      "[1] 8600\n",
      "[1] 8700\n",
      "[1] 8800\n",
      "[1] 8900\n",
      "[1] 9000\n",
      "[1] 9100\n",
      "[1] 9200\n",
      "[1] 9300\n",
      "[1] 9400\n",
      "[1] 9500\n",
      "[1] 9600\n",
      "[1] 9700\n",
      "[1] 9800\n",
      "[1] 9900\n",
      "[1] 10000\n",
      "[1] 10100\n",
      "[1] 10200\n",
      "[1] 10300\n",
      "[1] 10400\n",
      "[1] 10500\n",
      "[1] 10600\n",
      "[1] 10700\n",
      "[1] 10800\n",
      "[1] 10900\n",
      "[1] 11000\n",
      "[1] 11100\n",
      "[1] 11200\n"
     ]
    }
   ],
   "source": [
    "# generate javascript file for the variant proportion for each TCGA sample\n",
    "for (i in 1:nrow(all_tcga)){\n",
    "    filname_i <- paste0(\"./RNA_mod_for_each_sample/\",all_tcga[i, \"analysis_id\"])\n",
    "    sample_i <- read.table(filname_i, sep=\"\\t\", header = FALSE)\n",
    "    colnames(sample_i) <- c(\"analysis_id\", \"chrom\", \"position\",\n",
    "                           \"reference_nt\", \"variant_proportion\", \"depth\",\n",
    "                           \"deletionCount\", \"deletion_proportion\")\n",
    "    \n",
    "    sample_i_non0 <- sample_i[which(sample_i$variant_proportion > 0),]\n",
    "    \n",
    "    chr <-  sample_i_non0[,\"chrom\",drop=FALSE]\n",
    "    position <- sample_i_non0[,\"position\",drop=FALSE]\n",
    "    reference_nt <- sample_i_non0[,\"reference_nt\",drop=FALSE]\n",
    "    \n",
    "    x_max_round <- round(x = max(sample_i_non0$variant_proportion), digits = 1)\n",
    "    if (x_max_round >= max(sample_i_non0$variant_proportion)){\n",
    "        x_max <- x_max_round\n",
    "    } else {\n",
    "        x_max <- x_max_round + 0.05\n",
    "    }\n",
    "    p <- suppressWarnings(rbokeh::figure(xgrid=FALSE, ygrid = FALSE, xaxes = \"below\", yaxes = \"left\",\n",
    "              width = 500, height = 500, ylim=c(0, x_max),\n",
    "              logo = NULL, tools = c( \"pan\", \"wheel_zoom\",\"box_zoom\",\"reset\"), \n",
    "                                         xlab=\"\", ylab=\"variation proportion\") %>% \n",
    "    rbokeh::ly_points(x = seq(1,nrow(sample_i_non0),1), y=sample_i_non0$variant_proportion, col=\"red\",\n",
    "                         alpha=0.5, size=5, hover = list(chr, position, reference_nt)))\n",
    "    \n",
    "    \n",
    "    suppressMessages(suppressWarnings(rbokeh::rbokeh2html(p, file = paste0(\"./variation_prop_plot_html_for_modpatient/\",\n",
    "                                     all_tcga[i, \"analysis_id\"],\".html\"))))\n",
    "    \n",
    "    \n",
    "    index_html <- file(paste0(\"./variation_prop_plot_html_for_modpatient/\",\n",
    "                                     all_tcga[i, \"analysis_id\"],\".html\"),\"r\")\n",
    "    modelid = \"\"\n",
    "    docid = \"\"\n",
    "    docs_json = \"\"\n",
    "    while (TRUE){\n",
    "        line = readLines(index_html, n=1, warn = FALSE)\n",
    "        if ( length(line) == 0 ) {\n",
    "            break\n",
    "        }\n",
    "        line = trimws(line)\n",
    "        if(startsWith(line, \"var modelid\")){\n",
    "            modelid = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "        if(startsWith(line, \"var docid\")){\n",
    "            docid = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "        if(startsWith(line, \"var docs_json\")){\n",
    "            docs_json = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "    }\n",
    "    close(index_html)\n",
    "    fileConn <- file(paste0(\"./variation_prop_plot_html_data_for_modpatient/\",\n",
    "                            all_tcga[i, \"analysis_id\"],\".js\"))\n",
    "    writeLines(c(\"Bokeh.$(function() {\",\n",
    "                 modelid, \n",
    "                 paste0(\"var elementid = 'modpatient_modal_point_plot_\", \n",
    "                        all_tcga[i, \"analysis_id\"],\"';\"),\n",
    "                 docid, \n",
    "                 docs_json,\n",
    "                \"var refkey = Object.keys(docs_json)[0]\",\n",
    "                \"var refs = docs_json[refkey].roots.references\",\n",
    "                \"function traverseObject(obj) {\",\n",
    "                \"  for (var key in obj) {\",\n",
    "                \"    if (obj[key].constructor === Object) {\",\n",
    "                \"      traverseObject(obj[key]);\",\n",
    "                \"    } else if (obj[key].constructor === Array) {\",\n",
    "                \"      for (var i = 0; i < obj[key].length; i++) {\",\n",
    "                \"        if (obj[key][i] === null)\",\n",
    "                \"        obj[key][i] = NaN;\",\n",
    "                \"      };\",\n",
    "                \"    }\",\n",
    "                \"  };\",\n",
    "                \"}\",\n",
    "                \"for (var i = 0; i < refs.length; i++) {\",\n",
    "                \"  if (refs[i].type === 'ColumnDataSource')\",\n",
    "                \"    traverseObject(refs[i].attributes.data);\",\n",
    "                \"};\",\n",
    "                \"var render_items = [{\",\n",
    "                \"  'docid': docid,\",\n",
    "                \"  'elementid': elementid,\",\n",
    "                \"  'modelid': modelid\",\n",
    "                \"}];\",\n",
    "                \"Bokeh.set_log_level('info');\",\n",
    "                \"Bokeh.embed.embed_items(docs_json, render_items);\",\n",
    "                \"});\"), fileConn)\n",
    "    close(fileConn)\n",
    "    \n",
    "    if (i%%100==0){print(i)}\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 100\n",
      "[1] 200\n",
      "[1] 300\n",
      "[1] 400\n",
      "[1] 500\n",
      "[1] 600\n",
      "[1] 700\n",
      "[1] 800\n",
      "[1] 900\n",
      "[1] 1000\n",
      "[1] 1100\n",
      "[1] 1200\n",
      "[1] 1300\n",
      "[1] 1400\n",
      "[1] 1500\n",
      "[1] 1600\n",
      "[1] 1700\n",
      "[1] 1800\n",
      "[1] 1900\n",
      "[1] 2000\n",
      "[1] 2100\n",
      "[1] 2200\n",
      "[1] 2300\n",
      "[1] 2400\n",
      "[1] 2500\n",
      "[1] 2600\n",
      "[1] 2700\n",
      "[1] 2800\n",
      "[1] 2900\n",
      "[1] 3000\n",
      "[1] 3100\n",
      "[1] 3200\n",
      "[1] 3300\n",
      "[1] 3400\n",
      "[1] 3500\n",
      "[1] 3600\n",
      "[1] 3700\n",
      "[1] 3800\n",
      "[1] 3900\n",
      "[1] 4000\n",
      "[1] 4100\n",
      "[1] 4200\n",
      "[1] 4300\n",
      "[1] 4400\n",
      "[1] 4500\n",
      "[1] 4600\n",
      "[1] 4700\n",
      "[1] 4800\n",
      "[1] 4900\n",
      "[1] 5000\n",
      "[1] 5100\n",
      "[1] 5200\n",
      "[1] 5300\n",
      "[1] 5400\n",
      "[1] 5500\n",
      "[1] 5600\n",
      "[1] 5700\n",
      "[1] 5800\n",
      "[1] 5900\n",
      "[1] 6000\n",
      "[1] 6100\n",
      "[1] 6200\n",
      "[1] 6300\n",
      "[1] 6400\n",
      "[1] 6500\n",
      "[1] 6600\n",
      "[1] 6700\n",
      "[1] 6800\n",
      "[1] 6900\n",
      "[1] 7000\n",
      "[1] 7100\n",
      "[1] 7200\n",
      "[1] 7300\n",
      "[1] 7400\n",
      "[1] 7500\n",
      "[1] 7600\n",
      "[1] 7700\n",
      "[1] 7800\n",
      "[1] 7900\n",
      "[1] 8000\n",
      "[1] 8100\n",
      "[1] 8200\n",
      "[1] 8300\n",
      "[1] 8400\n",
      "[1] 8500\n",
      "[1] 8600\n",
      "[1] 8700\n",
      "[1] 8800\n",
      "[1] 8900\n",
      "[1] 9000\n",
      "[1] 9100\n",
      "[1] 9200\n",
      "[1] 9300\n",
      "[1] 9400\n",
      "[1] 9500\n",
      "[1] 9600\n",
      "[1] 9700\n",
      "[1] 9800\n",
      "[1] 9900\n",
      "[1] 10000\n",
      "[1] 10100\n",
      "[1] 10200\n",
      "[1] 10300\n",
      "[1] 10400\n",
      "[1] 10500\n",
      "[1] 10600\n",
      "[1] 10700\n",
      "[1] 10800\n",
      "[1] 10900\n",
      "[1] 11000\n",
      "[1] 11100\n",
      "[1] 11200\n"
     ]
    }
   ],
   "source": [
    "# generate javascript file for the deletion proportion for each TCGA sample\n",
    "for (i in 1:nrow(all_tcga)){\n",
    "    filname_i <- paste0(\"./RNA_mod_for_each_sample/\",all_tcga[i, \"analysis_id\"])\n",
    "    sample_i <- read.table(filname_i, sep=\"\\t\", header = FALSE)\n",
    "    colnames(sample_i) <- c(\"analysis_id\", \"chrom\", \"position\",\n",
    "                           \"reference_nt\", \"variant_proportion\", \"depth\",\n",
    "                           \"deletionCount\", \"deletion_proportion\")\n",
    "    \n",
    "    sample_i_non0 <- sample_i[which(sample_i$variant_proportion > 0),]\n",
    "    \n",
    "    chr <-  sample_i_non0[,\"chrom\",drop=FALSE]\n",
    "    position <- sample_i_non0[,\"position\",drop=FALSE]\n",
    "    reference_nt <- sample_i_non0[,\"reference_nt\",drop=FALSE]\n",
    "    \n",
    "    x_max_round <- round(x = max(sample_i_non0$deletion_proportion), digits = 1)\n",
    "    if (x_max_round >= max(sample_i_non0$deletion_proportion)){\n",
    "        x_max <- x_max_round\n",
    "    } else {\n",
    "        x_max <- x_max_round + 0.05\n",
    "    }\n",
    "    p <- suppressWarnings(rbokeh::figure(xgrid=FALSE, ygrid = FALSE, xaxes = \"below\", yaxes = \"left\",\n",
    "              width = 500, height = 500, ylim=c(0, x_max),\n",
    "              logo = NULL, tools = c( \"pan\", \"wheel_zoom\",\"box_zoom\",\"reset\"), \n",
    "                                         xlab=\"\", ylab=\"deletion proportion\") %>% \n",
    "    rbokeh::ly_points(x = seq(1,nrow(sample_i_non0),1), y=sample_i_non0$deletion_proportion, col=\"red\",\n",
    "                         alpha=0.5, size=5, hover = list(chr, position, reference_nt)))\n",
    "    \n",
    "    \n",
    "    suppressMessages(suppressWarnings(rbokeh::rbokeh2html(p, file = paste0(\"./deletion_prop_plot_html_for_modpatient/\",\n",
    "                                     all_tcga[i, \"analysis_id\"],\".html\"))))\n",
    "    \n",
    "    \n",
    "    index_html <- file(paste0(\"./deletion_prop_plot_html_for_modpatient/\",\n",
    "                                     all_tcga[i, \"analysis_id\"],\".html\"),\"r\")\n",
    "    modelid = \"\"\n",
    "    docid = \"\"\n",
    "    docs_json = \"\"\n",
    "    while (TRUE){\n",
    "        line = readLines(index_html, n=1, warn = FALSE)\n",
    "        if ( length(line) == 0 ) {\n",
    "            break\n",
    "        }\n",
    "        line = trimws(line)\n",
    "        if(startsWith(line, \"var modelid\")){\n",
    "            modelid = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "        if(startsWith(line, \"var docid\")){\n",
    "            docid = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "        if(startsWith(line, \"var docs_json\")){\n",
    "            docs_json = gsub(\"[\\r\\n]\", \"\", line)\n",
    "        }\n",
    "    }\n",
    "    close(index_html)\n",
    "    fileConn <- file(paste0(\"./deletion_prop_plot_html_data_for_modpatient/\",\n",
    "                            all_tcga[i, \"analysis_id\"],\".js\"))\n",
    "    writeLines(c(\"Bokeh.$(function() {\",\n",
    "                 modelid, \n",
    "                 paste0(\"var elementid = 'modpatient_modal_point_plot_deletion_\", \n",
    "                        all_tcga[i, \"analysis_id\"],\"';\"),\n",
    "                 docid, \n",
    "                 docs_json,\n",
    "                \"var refkey = Object.keys(docs_json)[0]\",\n",
    "                \"var refs = docs_json[refkey].roots.references\",\n",
    "                \"function traverseObject(obj) {\",\n",
    "                \"  for (var key in obj) {\",\n",
    "                \"    if (obj[key].constructor === Object) {\",\n",
    "                \"      traverseObject(obj[key]);\",\n",
    "                \"    } else if (obj[key].constructor === Array) {\",\n",
    "                \"      for (var i = 0; i < obj[key].length; i++) {\",\n",
    "                \"        if (obj[key][i] === null)\",\n",
    "                \"        obj[key][i] = NaN;\",\n",
    "                \"      };\",\n",
    "                \"    }\",\n",
    "                \"  };\",\n",
    "                \"}\",\n",
    "                \"for (var i = 0; i < refs.length; i++) {\",\n",
    "                \"  if (refs[i].type === 'ColumnDataSource')\",\n",
    "                \"    traverseObject(refs[i].attributes.data);\",\n",
    "                \"};\",\n",
    "                \"var render_items = [{\",\n",
    "                \"  'docid': docid,\",\n",
    "                \"  'elementid': elementid,\",\n",
    "                \"  'modelid': modelid\",\n",
    "                \"}];\",\n",
    "                \"Bokeh.set_log_level('info');\",\n",
    "                \"Bokeh.embed.embed_items(docs_json, render_items);\",\n",
    "                \"});\"), fileConn)\n",
    "    close(fileConn)\n",
    "    \n",
    "    if (i%%100==0){print(i)}\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ModPatient - Gene ontology enrichment of genes associated with RNA modification loci for each TCGA patient sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Welcome to enrichR\n",
      "Checking connection ... \n",
      "Connection is Live!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(enrichR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>analysis_id</th><th scope=col>barcode_final</th><th scope=col>project_id</th><th scope=col>barcode</th><th scope=col>disease</th><th scope=col>disease_name</th><th scope=col>tumor_tissue_site</th><th scope=col>histological_type</th><th scope=col>gender</th><th scope=col>days_to_initial_pathologic_diagnosis</th><th scope=col>age_at_initial_pathologic_diagnosis</th><th scope=col>vital_status</th><th scope=col>days_to_birth</th><th scope=col>days_to_death</th><th scope=col>days_to_last_followup</th><th scope=col>patient_death_reason</th><th scope=col>race_list</th><th scope=col>treatment</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>fb47df4a-b9a7-46c6-9e8d-247c00057185</td><td>TCGA-02-0047</td><td>TCGA-GBM</td><td>TCGA-02-0047-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>78</td><td>Dead </td><td>-28759</td><td> 448</td><td> 448</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>e1a99d0d-6b86-484d-8662-23c4378cde92</td><td>TCGA-02-0055</td><td>TCGA-GBM</td><td>TCGA-02-0055-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>FEMALE</td><td>0</td><td>62</td><td>Dead </td><td>-22798</td><td>  76</td><td>  76</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f</td><td>TCGA-02-2483</td><td>TCGA-GBM</td><td>TCGA-02-2483-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>43</td><td>Alive</td><td>-15964</td><td>  NA</td><td> 466</td><td>NA</td><td>ASIAN                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>f752e5d9-00a8-41f1-ae38-2b589f499088</td><td>TCGA-02-2485</td><td>TCGA-GBM</td><td>TCGA-02-2485-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>53</td><td>Alive</td><td>-19494</td><td>  NA</td><td> 470</td><td>NA</td><td>BLACK OR AFRICAN AMERICAN</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>50127941-3e32-46a5-956e-f55fbfc4080f</td><td>TCGA-02-2486</td><td>TCGA-GBM</td><td>TCGA-02-2486-01A-01R-1849-01</td><td>GBM</td><td>Glioblastoma multiforme          </td><td>Brain</td><td>Untreated primary (de novo) GBM</td><td>MALE  </td><td>0</td><td>64</td><td>Alive</td><td>-23394</td><td>  NA</td><td> 493</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>2e13491b-c707-49e6-852c-ccd1d6252b00</td><td>TCGA-04-1331</td><td>TCGA-OV </td><td>TCGA-04-1331-01A-01R-1569-13</td><td>OV </td><td>Ovarian serous cystadenocarcinoma</td><td>Ovary</td><td>Serous Cystadenocarcinoma      </td><td>FEMALE</td><td>0</td><td>78</td><td>Dead </td><td>-28848</td><td>1336</td><td>1224</td><td>NA</td><td>WHITE                    </td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 18\n",
       "\\begin{tabular}{r|llllllllllllllllll}\n",
       "  & analysis\\_id & barcode\\_final & project\\_id & barcode & disease & disease\\_name & tumor\\_tissue\\_site & histological\\_type & gender & days\\_to\\_initial\\_pathologic\\_diagnosis & age\\_at\\_initial\\_pathologic\\_diagnosis & vital\\_status & days\\_to\\_birth & days\\_to\\_death & days\\_to\\_last\\_followup & patient\\_death\\_reason & race\\_list & treatment\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <int> & <int> & <fct> & <int> & <int> & <int> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & fb47df4a-b9a7-46c6-9e8d-247c00057185 & TCGA-02-0047 & TCGA-GBM & TCGA-02-0047-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 78 & Dead  & -28759 &  448 &  448 & NA & WHITE                     & NA\\\\\n",
       "\t2 & e1a99d0d-6b86-484d-8662-23c4378cde92 & TCGA-02-0055 & TCGA-GBM & TCGA-02-0055-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & FEMALE & 0 & 62 & Dead  & -22798 &   76 &   76 & NA & WHITE                     & NA\\\\\n",
       "\t3 & 00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f & TCGA-02-2483 & TCGA-GBM & TCGA-02-2483-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 43 & Alive & -15964 &   NA &  466 & NA & ASIAN                     & NA\\\\\n",
       "\t4 & f752e5d9-00a8-41f1-ae38-2b589f499088 & TCGA-02-2485 & TCGA-GBM & TCGA-02-2485-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 53 & Alive & -19494 &   NA &  470 & NA & BLACK OR AFRICAN AMERICAN & NA\\\\\n",
       "\t5 & 50127941-3e32-46a5-956e-f55fbfc4080f & TCGA-02-2486 & TCGA-GBM & TCGA-02-2486-01A-01R-1849-01 & GBM & Glioblastoma multiforme           & Brain & Untreated primary (de novo) GBM & MALE   & 0 & 64 & Alive & -23394 &   NA &  493 & NA & WHITE                     & NA\\\\\n",
       "\t6 & 2e13491b-c707-49e6-852c-ccd1d6252b00 & TCGA-04-1331 & TCGA-OV  & TCGA-04-1331-01A-01R-1569-13 & OV  & Ovarian serous cystadenocarcinoma & Ovary & Serous Cystadenocarcinoma       & FEMALE & 0 & 78 & Dead  & -28848 & 1336 & 1224 & NA & WHITE                     & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 18\n",
       "\n",
       "| <!--/--> | analysis_id &lt;fct&gt; | barcode_final &lt;fct&gt; | project_id &lt;fct&gt; | barcode &lt;fct&gt; | disease &lt;fct&gt; | disease_name &lt;fct&gt; | tumor_tissue_site &lt;fct&gt; | histological_type &lt;fct&gt; | gender &lt;fct&gt; | days_to_initial_pathologic_diagnosis &lt;int&gt; | age_at_initial_pathologic_diagnosis &lt;int&gt; | vital_status &lt;fct&gt; | days_to_birth &lt;int&gt; | days_to_death &lt;int&gt; | days_to_last_followup &lt;int&gt; | patient_death_reason &lt;fct&gt; | race_list &lt;fct&gt; | treatment &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | fb47df4a-b9a7-46c6-9e8d-247c00057185 | TCGA-02-0047 | TCGA-GBM | TCGA-02-0047-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 78 | Dead  | -28759 |  448 |  448 | NA | WHITE                     | NA |\n",
       "| 2 | e1a99d0d-6b86-484d-8662-23c4378cde92 | TCGA-02-0055 | TCGA-GBM | TCGA-02-0055-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | FEMALE | 0 | 62 | Dead  | -22798 |   76 |   76 | NA | WHITE                     | NA |\n",
       "| 3 | 00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f | TCGA-02-2483 | TCGA-GBM | TCGA-02-2483-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 43 | Alive | -15964 |   NA |  466 | NA | ASIAN                     | NA |\n",
       "| 4 | f752e5d9-00a8-41f1-ae38-2b589f499088 | TCGA-02-2485 | TCGA-GBM | TCGA-02-2485-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 53 | Alive | -19494 |   NA |  470 | NA | BLACK OR AFRICAN AMERICAN | NA |\n",
       "| 5 | 50127941-3e32-46a5-956e-f55fbfc4080f | TCGA-02-2486 | TCGA-GBM | TCGA-02-2486-01A-01R-1849-01 | GBM | Glioblastoma multiforme           | Brain | Untreated primary (de novo) GBM | MALE   | 0 | 64 | Alive | -23394 |   NA |  493 | NA | WHITE                     | NA |\n",
       "| 6 | 2e13491b-c707-49e6-852c-ccd1d6252b00 | TCGA-04-1331 | TCGA-OV  | TCGA-04-1331-01A-01R-1569-13 | OV  | Ovarian serous cystadenocarcinoma | Ovary | Serous Cystadenocarcinoma       | FEMALE | 0 | 78 | Dead  | -28848 | 1336 | 1224 | NA | WHITE                     | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  analysis_id                          barcode_final project_id\n",
       "1 fb47df4a-b9a7-46c6-9e8d-247c00057185 TCGA-02-0047  TCGA-GBM  \n",
       "2 e1a99d0d-6b86-484d-8662-23c4378cde92 TCGA-02-0055  TCGA-GBM  \n",
       "3 00b3b72b-0ac4-4016-bfa2-0bfcd5645d4f TCGA-02-2483  TCGA-GBM  \n",
       "4 f752e5d9-00a8-41f1-ae38-2b589f499088 TCGA-02-2485  TCGA-GBM  \n",
       "5 50127941-3e32-46a5-956e-f55fbfc4080f TCGA-02-2486  TCGA-GBM  \n",
       "6 2e13491b-c707-49e6-852c-ccd1d6252b00 TCGA-04-1331  TCGA-OV   \n",
       "  barcode                      disease disease_name                     \n",
       "1 TCGA-02-0047-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "2 TCGA-02-0055-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "3 TCGA-02-2483-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "4 TCGA-02-2485-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "5 TCGA-02-2486-01A-01R-1849-01 GBM     Glioblastoma multiforme          \n",
       "6 TCGA-04-1331-01A-01R-1569-13 OV      Ovarian serous cystadenocarcinoma\n",
       "  tumor_tissue_site histological_type               gender\n",
       "1 Brain             Untreated primary (de novo) GBM MALE  \n",
       "2 Brain             Untreated primary (de novo) GBM FEMALE\n",
       "3 Brain             Untreated primary (de novo) GBM MALE  \n",
       "4 Brain             Untreated primary (de novo) GBM MALE  \n",
       "5 Brain             Untreated primary (de novo) GBM MALE  \n",
       "6 Ovary             Serous Cystadenocarcinoma       FEMALE\n",
       "  days_to_initial_pathologic_diagnosis age_at_initial_pathologic_diagnosis\n",
       "1 0                                    78                                 \n",
       "2 0                                    62                                 \n",
       "3 0                                    43                                 \n",
       "4 0                                    53                                 \n",
       "5 0                                    64                                 \n",
       "6 0                                    78                                 \n",
       "  vital_status days_to_birth days_to_death days_to_last_followup\n",
       "1 Dead         -28759         448           448                 \n",
       "2 Dead         -22798          76            76                 \n",
       "3 Alive        -15964          NA           466                 \n",
       "4 Alive        -19494          NA           470                 \n",
       "5 Alive        -23394          NA           493                 \n",
       "6 Dead         -28848        1336          1224                 \n",
       "  patient_death_reason race_list                 treatment\n",
       "1 NA                   WHITE                     NA       \n",
       "2 NA                   WHITE                     NA       \n",
       "3 NA                   ASIAN                     NA       \n",
       "4 NA                   BLACK OR AFRICAN AMERICAN NA       \n",
       "5 NA                   WHITE                     NA       \n",
       "6 NA                   WHITE                     NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "all_tcga <- read.table(\"tcga_ccle_samples.barcode.with_targeted_analysis_ids.with_clinical_info.for_db.txt\",\n",
    "                      sep=\"\\t\", header=TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# GO enrichment of genes associated with RNA modification loci (variantion proportion >= 0.01) for each TCGA sample\n",
    "\n",
    "loci_ann <- read.delim(\"all_genomic_loci_uniq_with_ids_and_annotations.txt\", \n",
    "                      sep=\"\\t\", header=FALSE)\n",
    "\n",
    "for (i in 1:nrow(all_tcga)){\n",
    "    filname_i <- paste0(\"./RNA_mod_for_each_sample/\",all_tcga[i, \"analysis_id\"])\n",
    "    sample_i <- read.table(filname_i, sep=\"\\t\", header = FALSE)\n",
    "    colnames(sample_i) <- c(\"analysis_id\", \"chrom\", \"position\",\n",
    "                           \"reference_nt\", \"variant_proportion\", \"depth\",\n",
    "                           \"deletionCount\", \"deletion_proportion\")\n",
    "    sample_i_non0 <- sample_i[which(sample_i$variant_proportion >= 0.01 ),]\n",
    "    sample_i_non0$id <- paste(sample_i_non0$chrom, sample_i_non0$position, sample_i_non0$reference_nt, sep = \"-\")\n",
    "    \n",
    "    loci_ann_i <- loci_ann[which(loci_ann$V1 %in% sample_i_non0$id),]\n",
    "    \n",
    "    dbs <- \"GO_Biological_Process_2018\"\n",
    "    enriched_i <- enrichr(as.character(unique(loci_ann_i$V6)), dbs)\n",
    "    \n",
    "    \n",
    "    enriched_i_BP <- enriched_i$GO_Biological_Process_2018\n",
    "    if (nrow(enriched_i_BP) > 0){\n",
    "        enriched_i_BP_sig <- enriched_i_BP[which(enriched_i_BP$Adjusted.P.value < 0.05), \n",
    "                                           c(\"Term\",\"Adjusted.P.value\", \"Genes\")]\n",
    "    } else {\n",
    "        enriched_i_BP_sig <- data.frame(Term=character(), \n",
    "                                        Adjusted.P.value=character(), \n",
    "                                        Genes=character())\n",
    "    }\n",
    "    \n",
    "    enriched_i_BP_filename <- paste0(\"./GO_enrichments_of_genes_for_modpatient/\",\n",
    "                                    all_tcga[i, \"analysis_id\"])\n",
    "    write.table(enriched_i_BP_sig, enriched_i_BP_filename, sep = \"\\t\",\n",
    "               row.names=FALSE, col.names=TRUE, quote=FALSE)\n",
    "    \n",
    "}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
